// carbon/lib/components/form/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _index = require('./flavors/input/index');

var _index2 = require('./flavors/input-search/index');

var _index3 = _interopRequireDefault(_index2);

var _index4 = require('./flavors/input-required/index');

var _index5 = _interopRequireDefault(_index4);

var _index6 = require('./flavors/textarea/index');

var _index7 = require('./flavors/textarea-required/index');

var _index8 = _interopRequireDefault(_index7);

var _index9 = require('./flavors/radio/index');

var _index10 = require('./flavors/radio-group-alternate/index');

var _index11 = require('./flavors/checkbox-group-alertnate/index');

var _index12 = require('./flavors/checkbox-toggle/index');

var _index13 = _interopRequireDefault(_index12);

var _index14 = require('./flavors/checkbox/index');

var _index15 = require('./flavors/checkbox-add-button/index');

var _index16 = _interopRequireDefault(_index15);

var _index17 = require('./flavors/select/index');

var _index18 = _interopRequireDefault(_index17);

var _index19 = require('./flavors/combox-select/index');

var _index20 = _interopRequireDefault(_index19);

var _index21 = require('./flavors/form-multi-select/index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * https://www.lightningdesignsystem.com/components/forms/
 * 1. 输入框
 * 2. 文本输入区域框
 * 3. 单选项
 * 4. 单选组
 * 5. 复选项
 * 6. 切换开关
 * 7. 复选组
 * 8. 单选选择器
 * 9. 多选选择器
 * 10. 水平垂直排布表单
 * 11. 标准排布表单
 * 12. 单行排布表单
 * 13. 混合排布表单
 * 14. 表单标签提示
 * 15. 表单固定底部
 * 16. 两栏表单
 */
exports.default = {
    Input: _index.Input,
    DisabledInput: _index.DisabledInput,
    ReadOnlyInput: _index.ReadOnlyInput,
    ReadOnlyFixedText: _index.ReadOnlyFixedText,
    FixedInput: _index.FixedInput,
    InputSearch: _index3.default,
    InputRequird: _index5.default,
    TextArea: _index6.TextArea,
    DisabledTextArea: _index6.DisabledTextArea,
    ReadOnlyTextArea: _index6.ReadOnlyTextArea,
    TextAreaRequired: _index8.default,
    Radio: _index9.Radio,
    RadioGroup: _index9.RadioGroup,
    RequiredRadioGroup: _index9.RequiredRadioGroup,
    AlternateRadio: _index10.AlternateRadio,
    AlternateGroup: _index10.AlternateGroup,
    Checkbox: _index14.Checkbox,
    CheckboxGroup: _index14.CheckboxGroup,
    RequiredCheckboxGroup: _index14.RequiredCheckboxGroup,
    CheckboxToggle: _index13.default,
    AlternateCheckbox: _index11.AlternateCheckbox,
    AlternateCheckboxGroup: _index11.AlternateCheckboxGroup,
    CheckboxAddButton: _index16.default,
    Select: _index18.default,
    ComboxSelect: _index20.default,
    FromMultiSelect: _index21.FromMultiSelect
};
module.exports = exports['default'];
