// carbon/lib/components/button-group/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _index = require('./index');

var _index2 = require('./../buttons/index');

var _index3 = require('./../button-icon/flavors/base/index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ReactElementToString = require('react-element-to-string');

var Demo = function Demo(_ref) {
    var children = _ref.children;
    return _react2.default.createElement(
        'section',
        { className: 'component' },
        _react2.default.createElement(
            'h1',
            null,
            '\u6309\u94AE\u7EC4'
        ),
        _react2.default.createElement(
            'div',
            { className: 'example' },
            _react2.default.createElement(
                'div',
                { className: 'example-content' },
                children
            )
        )
    );
};

Demo.propTypes = {
    children: _propTypes2.default.any
};

var buttonGroup = _react2.default.createElement(
    _index.ButtonGroup,
    { className: 'aaa' },
    _react2.default.createElement(
        _index2.Button,
        { className: 'mcds-button__item', active: true },
        'Refresh'
    ),
    _react2.default.createElement(
        _index2.Button,
        { className: 'mcds-button__item' },
        'Edit'
    ),
    _react2.default.createElement(
        _index3.ButtonIcon,
        { className: 'mcds-button__item', icon: 'mcds-icon__triangle-solid-14' },
        'Save'
    )
);

var ButtonGroupDemo = function ButtonGroupDemo() {
    return _react2.default.createElement(
        Demo,
        null,
        buttonGroup
    );
};

exports.default = [{
    id: 'ButtonGroup',
    element: _react2.default.createElement(ButtonGroupDemo, null),
    code: ReactElementToString(buttonGroup)
}];
module.exports = exports['default'];
