// carbon/lib/components/modal/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkdown = require('react-markdown');

var _reactMarkdown2 = _interopRequireDefault(_reactMarkdown);

var _index = require('./index');

var _index2 = require('../buttons/index');

var _dropdown = require('../dropdown');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ModelDefault = _react2.default.createElement(
    _index.ModalTrigger,
    null,
    _react2.default.createElement(
        _index2.Button,
        { className: 'mcds-button__brand' },
        'Modal'
    ),
    _react2.default.createElement(
        _index.Modal,
        { className: 'mcds-modal__w-820 mcds-modal__auto' },
        _react2.default.createElement(
            _index.ModalHeader,
            null,
            _react2.default.createElement('i', { className: 'mcds-modal__close mcds-icon__close-line-20 close' }),
            _react2.default.createElement(
                'p',
                { className: 'mcds-modal__title' },
                '\u65B0\u5EFA'
            )
        ),
        _react2.default.createElement(
            _index.ModalBody,
            null,
            _react2.default.createElement(
                'p',
                null,
                'This component is essentially 2 ARIA listboxes side by side, so we follow the [ARIA practices guide]'
            )
        ),
        _react2.default.createElement(
            _index.ModalFoot,
            null,
            _react2.default.createElement(
                'div',
                { className: 'mcds-layout__column mcds-layout__right' },
                _react2.default.createElement(
                    _index2.Button,
                    { className: 'mcds-button__neutral mcds-btn__right close' },
                    'Cancel'
                ),
                _react2.default.createElement(
                    _index2.Button,
                    { className: 'mcds-button__brand close' },
                    'Save'
                )
            )
        )
    )
); /**
    * 存放示例
    */

var ModelStateClose = function (_React$Component) {
    (0, _inherits3.default)(ModelStateClose, _React$Component);

    function ModelStateClose() {
        (0, _classCallCheck3.default)(this, ModelStateClose);

        var _this = (0, _possibleConstructorReturn3.default)(this, (ModelStateClose.__proto__ || (0, _getPrototypeOf2.default)(ModelStateClose)).call(this));

        _this.state = { active: false };
        return _this;
    }

    (0, _createClass3.default)(ModelStateClose, [{
        key: 'test',
        value: function test() {
            this.setState({ active: !this.state.active });
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(
                    'button',
                    null,
                    '123'
                ),
                _react2.default.createElement(
                    _dropdown.DropDownTrigger,
                    { target: 'body', autoCloseTag: 'ccc', placement: 'bottom-left' },
                    _react2.default.createElement(
                        _dropdown.DropDownItem,
                        null,
                        '\u5220\u9664'
                    ),
                    _react2.default.createElement(
                        _dropdown.DropDown,
                        { className: 'mcds-dropdown__min-no' },
                        _react2.default.createElement(
                            _dropdown.DropDownList,
                            null,
                            _react2.default.createElement(
                                _index.ModalTrigger,
                                null,
                                _react2.default.createElement(
                                    _dropdown.DropDownItem,
                                    { className: 'ccc' },
                                    '\u5220\u9664'
                                ),
                                _react2.default.createElement(
                                    _index.Modal,
                                    { className: 'mcds-modal__w-820' },
                                    _react2.default.createElement(
                                        _index.ModalHeader,
                                        null,
                                        _react2.default.createElement('i', { className: 'mcds-modal__close mcds-icon__close-line-20 close' }),
                                        _react2.default.createElement(
                                            'p',
                                            { className: 'mcds-modal__title' },
                                            '\u65B0\u5EFA'
                                        )
                                    ),
                                    _react2.default.createElement(
                                        _index.ModalBody,
                                        null,
                                        _react2.default.createElement(
                                            'p',
                                            null,
                                            'This component is essentially 2 ARIA listboxes side by side, so we follow the [ARIA practices guide]'
                                        )
                                    ),
                                    _react2.default.createElement(
                                        _index.ModalFoot,
                                        null,
                                        _react2.default.createElement(
                                            'div',
                                            { className: 'mcds-layout__column mcds-layout__right' },
                                            _react2.default.createElement(
                                                _index2.Button,
                                                { className: 'mcds-button__neutral mcds-btn__right close' },
                                                'Cancel'
                                            ),
                                            _react2.default.createElement(
                                                _index2.Button,
                                                { className: 'mcds-button__brand close' },
                                                'Save'
                                            )
                                        )
                                    )
                                )
                            )
                        )
                    )
                ),
                _react2.default.createElement(
                    _index2.Button,
                    { className: 'mcds-button__brand', onClick: this.test.bind(this) },
                    'Modal'
                ),
                _react2.default.createElement(
                    _index.Modal,
                    { show: this.state.active },
                    _react2.default.createElement(
                        _index.ModalHeader,
                        null,
                        _react2.default.createElement('i', { className: 'mcds-modal__close mcds-icon__close-line-20', onClick: this.test.bind(this) }),
                        _react2.default.createElement(
                            'p',
                            { className: 'mcds-modal__title' },
                            'header'
                        )
                    ),
                    _react2.default.createElement(
                        _index.ModalBody,
                        null,
                        _react2.default.createElement(
                            'p',
                            null,
                            'This component is essentially 2 ARIA listboxes side by side, so we follow the [ARIA practices guide]'
                        )
                    ),
                    _react2.default.createElement(
                        _index.ModalFoot,
                        null,
                        _react2.default.createElement(
                            'div',
                            { className: 'mcds-layout__column mcds-layout__right' },
                            _react2.default.createElement(
                                _index2.Button,
                                { className: 'mcds-button__neutral mcds-btn__right', onClick: this.test.bind(this) },
                                'Cancel'
                            ),
                            _react2.default.createElement(
                                _index2.Button,
                                { className: 'mcds-button__brand', onClick: this.test.bind(this) },
                                'Save'
                            )
                        )
                    )
                )
            );
        }
    }]);
    return ModelStateClose;
}(_react2.default.Component);

var ModelCode = '\nclass ModelStateClose extends React.Component {\n    ...\n    constructor() {\n        super();\n        this.state = { active: false };\n    }\n    test() {\n        this.setState({active: !this.state.active});\n    }\n    render() {\n        return (\n            <div>\n                <Button className="mcds-button__brand" onClick={this.test.bind(this)}>Modal</Button>\n                <Modal show={this.state.active} className="mcds-modal__content">\n                    <ModalHeader>\n                        <i className="mcds-modal__close mcds-icon__close-medium-14" onClick={this.test.bind(this)} />\n                        <p className="mcds-modal__title">header</p>\n                    </ModalHeader>\n                    <ModalBody>\n                        <p>This component is essentially 2 ARIA listboxes side by side, so we follow the [ARIA practices guide]</p>\n                    </ModalBody>\n\n                    <ModalFoot>\n                        <Button className="mcds-button__neutral mcds-btn__right" onClick={this.test.bind(this)}>\n                            Cancel\n                        </Button>\n                        <Button className="mcds-button__brand" onClick={this.test.bind(this)}>\n                            Save\n                        </Button>\n                    </ModalFoot>\n                </Modal>\n            </div>\n        );\n    }\n}';

exports.default = [{
    id: 'ModalTrigger',
    element: ModelDefault,
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: '#### ModalTrigger\r\n* ModalTrigger 第一个元素用来触发Modal 第二个元素为Modal\r\n* 宽度有两种:mcds-modal__w-520 mcds-modal__w-820\r\n* 全屏居中: mcds-modal__auto\r\n* 如果想点击按钮关闭Modal 只需给该按钮添加close样式即可，其他按钮同理\r\n* 如果不想通过增加class样式来关闭Modal 也可以通过操作active来关闭' })
}, {
    id: 'ModelStateClose',
    element: _react2.default.createElement(ModelStateClose, null),
    code: ModelCode,
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: '#### Modal\r\n* modal支持传入以下参数 show = null, trigger = null, closeOnEsc = true, onOpen, onClose, 如果是通过trigger触发,最后在onClose回调加入setState\r\n* 宽度有两种:mcds-modal__w-520 mcds-modal__w-820\r\n* 如果想点击按钮关闭Modal 只需给该按钮添加close样式即可，其他按钮同理\r\n* 如果不想通过增加class样式来关闭Modal 也可以通过操作active来关闭' })
}];
module.exports = exports['default'];
