// carbon/lib/components/modal/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _portal = require('../../base-components/portal');

var _portal2 = _interopRequireDefault(_portal);

var _dom = require('../../utils/dom');

var _dom2 = _interopRequireDefault(_dom);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Modal = function (_React$Component) {
    (0, _inherits3.default)(Modal, _React$Component);

    function Modal(props) {
        (0, _classCallCheck3.default)(this, Modal);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Modal.__proto__ || (0, _getPrototypeOf2.default)(Modal)).call(this, props));

        _this.state = {
            id: Math.random().toString(36).substring(7)
        };
        _this.closeModal = _this.closeModal.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(Modal, [{
        key: 'closeModal',
        value: function closeModal() {
            this.fade();
        }
    }, {
        key: '_handleOnOpen',
        value: function _handleOnOpen() {
            this.show();
        }
    }, {
        key: 'fade',
        value: function fade() {
            var _this2 = this;

            var modal = document.getElementById(this.state.id);
            _dom2.default.addClass(modal, 'mcds-modal__fade');
            _dom2.default.removeClass(modal, 'mcds-modal__in');
            setTimeout(function () {
                if (_this2 && _this2.node && _this2.node.closePortal) {
                    _this2.node.closePortal();
                }
            }, 300);
        }
    }, {
        key: 'show',
        value: function show() {
            var modal = document.getElementById(this.state.id);
            setTimeout(function () {
                _dom2.default.removeClass(modal, 'mcds-modal__fade');
                _dom2.default.addClass(modal, 'mcds-modal__in');
            }, 100);
        }
    }, {
        key: 'render',
        value: function render() {
            var _this3 = this;

            var _props = this.props,
                show = _props.show,
                _props$trigger = _props.trigger,
                trigger = _props$trigger === undefined ? null : _props$trigger,
                onOpen = _props.onOpen,
                _props$closeOnEsc = _props.closeOnEsc,
                closeOnEsc = _props$closeOnEsc === undefined ? true : _props$closeOnEsc,
                onClose = _props.onClose,
                others = (0, _objectWithoutProperties3.default)(_props, ['show', 'trigger', 'onOpen', 'closeOnEsc', 'onClose']);

            var classNames = (0, _classnames2.default)('mcds-modal', others.className);
            return _react2.default.createElement(
                _portal2.default,
                {
                    isOpened: show,
                    openByClickOn: trigger,
                    closeOnOutsideClick: false,
                    className: 'mcds-modal__portal',
                    closeOnEsc: closeOnEsc,
                    onOpen: this._handleOnOpen.bind(this),
                    onClose: onClose,
                    ref: function ref(node) {
                        _this3.node = node;
                    } },
                _react2.default.createElement(
                    'div',
                    (0, _extends3.default)({}, others, { id: this.state.id, className: classNames }),
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-modal__dialog' },
                        _react2.default.createElement(
                            'div',
                            { className: 'mcds-modal__content' },
                            others.children
                        )
                    )
                )
            );
        }
    }]);
    return Modal;
}(_react2.default.Component); /**
                               * https:// www.lightningdesignsystem.com/components/modals/
                               * 弹出框
                               */
/* eslint-disable */


Modal.propTypes = {
    show: _propTypes2.default.bool, // 是否显示
    trigger: _propTypes2.default.element, // 如果不设置show, 可以通过trigger按钮控制展示与否.他不可以和show一同使用
    children: _propTypes2.default.any,
    className: _propTypes2.default.string,
    closeOnEsc: _propTypes2.default.bool, // 是否esc按键关闭
    onOpen: _propTypes2.default.func, // 在打开的时候回调
    onClose: _propTypes2.default.func // 关闭的回调
};

Modal.defaultProps = {
    onOpen: function onOpen() {},
    onClose: function onClose() {}
};

var ModalTrigger = function (_React$Component2) {
    (0, _inherits3.default)(ModalTrigger, _React$Component2);

    function ModalTrigger() {
        (0, _classCallCheck3.default)(this, ModalTrigger);
        return (0, _possibleConstructorReturn3.default)(this, (ModalTrigger.__proto__ || (0, _getPrototypeOf2.default)(ModalTrigger)).apply(this, arguments));
    }

    (0, _createClass3.default)(ModalTrigger, [{
        key: 'componentWillMount',
        value: function componentWillMount() {
            try {
                var counts = _react2.default.Children.count(this.props.children);
                if (counts !== 2) {
                    throw new Error('ModalTrigger children must be two,like\r\n<ModalTrigger>\r\n\t<any>click</any>\r\n\t<Modal />\r\n</ModalTrigger>');
                }
            } catch (e) {
                console.error(e);
            }
        }
    }, {
        key: '_handleToggle',
        value: function _handleToggle(e) {
            if (_dom2.default.hasClass(e.target, 'close')) {
                this.modal.closeModal();
            }
        }
    }, {
        key: 'renderTrigger',
        value: function renderTrigger() {
            var children = _react2.default.Children.toArray(this.props.children);
            if (!children[0]) {
                return false;
            }
            var firstChild = _react2.default.cloneElement(children[0]);
            return firstChild;
        }
    }, {
        key: 'renderModal',
        value: function renderModal() {
            var _this5 = this;

            var children = _react2.default.Children.toArray(this.props.children);
            if (!children[1]) {
                return false;
            }
            var className = (0, _classnames2.default)(children[1].props.className);
            var secondChild = _react2.default.cloneElement(children[1], { className: className, ref: function ref(modal) {
                    _this5.modal = modal;
                }, trigger: this.renderTrigger(), onClick: this._handleToggle.bind(this) });
            return secondChild;
        }
    }, {
        key: 'render',
        value: function render() {
            return this.renderModal();
        }
    }]);
    return ModalTrigger;
}(_react2.default.Component);

ModalTrigger.propTypes = {
    children: _propTypes2.default.any
};

var ModalHeader = function ModalHeader(_ref) {
    var className = _ref.className,
        children = _ref.children,
        others = (0, _objectWithoutProperties3.default)(_ref, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-modal__header', className) }),
        children
    );
};

ModalHeader.propTypes = {
    children: _propTypes2.default.any,
    className: _propTypes2.default.string
};

var ModalBody = function ModalBody(_ref2) {
    var className = _ref2.className,
        children = _ref2.children,
        others = (0, _objectWithoutProperties3.default)(_ref2, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-modal__body', className) }),
        children
    );
};

ModalBody.propTypes = {
    children: _propTypes2.default.any,
    className: _propTypes2.default.string
};

var ModalFoot = function ModalFoot(_ref3) {
    var className = _ref3.className,
        children = _ref3.children,
        others = (0, _objectWithoutProperties3.default)(_ref3, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-modal__footer', className) }),
        children
    );
};

ModalFoot.propTypes = {
    children: _propTypes2.default.any,
    className: _propTypes2.default.string
};

exports.default = {
    ModalTrigger: ModalTrigger,
    Modal: Modal,
    ModalHeader: ModalHeader,
    ModalBody: ModalBody,
    ModalFoot: ModalFoot
};
module.exports = exports['default'];
