// carbon/lib/components/data-table/flavors/advanced/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Th = function (_React$Component) {
    (0, _inherits3.default)(Th, _React$Component);

    function Th(props) {
        (0, _classCallCheck3.default)(this, Th);
        return (0, _possibleConstructorReturn3.default)(this, (Th.__proto__ || (0, _getPrototypeOf2.default)(Th)).call(this, props));
    }

    (0, _createClass3.default)(Th, [{
        key: 'componentDidMount',
        value: function componentDidMount() {

            var th = this.refs.th;
            var cell = this.refs.cell;

            // 初始化的时候看上层组件有没有计算好宽度传递过来，如果有计算好的直接可以使用外层的宽度
            var width = window.getComputedStyle(th).width;

            if (this.props['data-columnWidth']) {
                width = this.props['data-columnWidth'] + 'px';
            }

            th.style.width = width;
            cell.style.width = width;
        }
    }, {
        key: 'render',
        value: function render() {
            var _props = this.props,
                children = _props.children,
                className = _props.className,
                onClick = _props.onClick,
                icon = _props.icon,
                resizable = _props.resizable,
                onMouseDown = _props.onMouseDown,
                onMouseUp = _props.onMouseUp,
                others = (0, _objectWithoutProperties3.default)(_props, ['children', 'className', 'onClick', 'icon', 'resizable', 'onMouseDown', 'onMouseUp']);


            return _react2.default.createElement(
                'th',
                (0, _extends3.default)({ ref: 'th', className: (0, _classnames2.default)('mcds-table__Th', className) }, others),
                _react2.default.createElement(
                    'div',
                    { ref: 'cell', className: 'mcds-table__resize-cell' },
                    _react2.default.createElement(
                        'a',
                        { className: 'mcds-text__weak mcds-table__resize-action', onClick: onClick },
                        _react2.default.createElement(
                            'span',
                            { className: 'mcds-truncate' },
                            children
                        ),
                        icon ? _react2.default.createElement('span', { className: (0, _classnames2.default)('mcds-table__icon mcds-icon__right mcds-icon__height', icon) }) : null
                    ),
                    resizable ? _react2.default.createElement('span', { className: 'mcds-resizable', onMouseDown: onMouseDown, onMouseUp: onMouseUp }) : null
                )
            );
        }
    }]);
    return Th;
}(_react2.default.Component);

Th.propTypes = {
    children: _propTypes2.default.any,
    icon: _propTypes2.default.string,
    resizable: _propTypes2.default.bool,
    onClick: _propTypes2.default.func,
    onMouseDown: _propTypes2.default.func,
    onMouseUp: _propTypes2.default.func,
    className: _propTypes2.default.string,
    'data-left': _propTypes2.default.number,
    'data-columnWidth': _propTypes2.default.number // 单元格宽度
};

Th.defaultProps = {
    'data-columnWidth': 0,
    onClick: function onClick() {}
};

exports.default = {
    Th: Th
};
module.exports = exports['default'];
