// carbon/lib/components/cascader/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkdown = require('react-markdown');

var _reactMarkdown2 = _interopRequireDefault(_reactMarkdown);

var _index = require('./index');

var _singleSelect = require('./single-select');

var _singleSelect2 = _interopRequireDefault(_singleSelect);

var _index2 = require('../buttons/index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import classnames from 'classnames';
var exampleData = [{
    name: 'one',
    value: '0',
    children: [{
        name: 'one-child',
        value: '0-1',
        children: [{
            name: 'one-son',
            value: '0-1-0'
        }]
    }, {
        name: 'two-child',
        value: '0-2',
        children: [{
            name: 'two-son',
            value: '0-2-0',
            children: [{
                name: 'one-child-son',
                value: '0-2-0-0',
                children: [{
                    name: 'one-son-son',
                    value: '0-2-0-0-0',
                    children: [{
                        name: 'one-child-son-son',
                        value: '0-2-0-0-0-0',
                        children: [{
                            name: 'one-son-son-son',
                            value: '0-2-0-0-0-0-0',
                            children: [{
                                name: 'one-child-son-son-son',
                                value: '0-2-0-0-0-0-0-0'
                            }, {
                                name: 'two-child-son-son-son',
                                value: '0-2-0-0-0-0-0-1'
                            }]
                        }]
                    }]
                }]
            }]
        }]
    }, {
        name: 'one-child',
        value: '0-3',
        children: [{
            name: 'two-son',
            value: '0-3-0',
            children: [{
                name: 'one-child-son',
                value: '0-3-0-0',
                children: [{
                    name: 'one-son-son',
                    value: '0-3-0-0-0',
                    children: [{
                        name: 'one-child-son-son',
                        value: '0-3-0-0-0-0',
                        children: [{
                            name: 'one-son-son-son',
                            value: '0-3-0-0-0-0-0',
                            children: [{
                                name: 'one-child-son-son-son',
                                value: '0-3-0-0-0-0-0-0'
                            }, {
                                name: 'two-child-son-son-son',
                                value: '0-3-0-0-0-0-0-1'
                            }]
                        }]
                    }]
                }]
            }]
        }]
    }, {
        name: 'two-child',
        value: '0-4'
    }, {
        name: 'one-child',
        value: '0-5'
    }, {
        name: 'two-child',
        value: '0-6'
    }, {
        name: 'one-child',
        value: '0-1'
    }, {
        name: 'two-child',
        value: '0-2'
    }, {
        name: 'one-child',
        value: '0-1'
    }, {
        name: 'two-child',
        value: '0-2'
    }, {
        name: 'one-child',
        value: '0-1',
        children: [{
            name: 'two-son',
            value: '1-2-0'
        }]
    }, {
        name: 'two-child',
        value: '0-2',
        children: [{
            name: 'two-son',
            value: '1-2-0'
        }]
    }, {
        name: 'one-child',
        value: '0-1',
        children: [{
            name: 'two-son',
            value: '1-2-0'
        }]
    }, {
        name: 'two-child',
        value: '0-2'
    }]
}, {
    name: 'two',
    value: '1',
    children: [{
        name: 'one-child',
        value: '1-1',
        children: [{
            name: 'two-son',
            value: '1-1-0'
        }]
    }, {
        name: 'two-child',
        value: '1-2',
        children: [{
            name: 'two-son',
            value: '2-2-0'
        }]
    }]
}, {
    name: '特斯轼',
    value: 't'
}];
// import {DropDown, DropDownList, DropDownItem, DropDownItemDivider, DropDownItemHeader} from '../dropdown';

// import {InputSearch} from '../index';

var defaultValue = ['0-3-0-0-0-0-0-0', '0-2-0-0-0-0-0-1'];

var MultiCascader = function MultiCascader() {
    return _react2.default.createElement(
        _index.Cascader,
        {
            error: true,
            data: exampleData,
            defaultValue: defaultValue,
            onChange: function onChange(v) {
                console.log('multi change', v);
            },
            type: 'multi',
            placement: 'right' },
        _react2.default.createElement(
            _index2.Button,
            { className: 'mcds-button__brand' },
            '\u591A\u9009\u62E9'
        )
    );
};

var SingleCascaderExample = '\nclass Example extends React.Component { \n    ...\n    render() {\n       const  exampleData = [\n            {\n                name:\'one\',\n                value:\'0\',\n                children:[\n                    {\n                        name:\'one-child\',\n                        value:\'0-1\'\n                    },\n                     {\n                        name:\'two-child\',\n                        value:\'0-2\'\n                    }\n                ]\n            },\n            {\n                name:\'\u7279\u65AF\u8F7C\',\n                value:\'t\'\n            }\n        ];\n        return (\n            <Cascader data={exampleData} defaultValue={[\'t\']} onChange={(v) => {console.log(\'multi change\',v);}} type=\'multi\' placement=\'right\'>\n                <Button className="mcds-button__brand">\u5355\u9879\u9009\u62E9</Button>\n            </Cascader>; \n        );\n    }\n}';
var MultiCascaderExample = '\nclass Example extends React.Component { \n    ...\n    render() {\n        const exampleData = [\n            {\n                name: \'one\',\n                value: \'0\',\n                children: [\n                    {\n                        name: \'one-child\',\n                        value: \'0-1\',\n                        children: [\n                            {\n                                name: \'one-son\',\n                                value: \'0-1-0\'\n                            }\n                        ]\n                    },\n                    {\n                        name: \'two-child\',\n                        value: \'0-2\',\n                        children: [\n                            {\n                                name: \'two-son\',\n                                value: \'0-2-0\',\n                                children: [\n                                    {\n                                        name: \'one-child-son\',\n                                        value: \'0-2-0-0\',\n                                        children: [\n                                            {\n                                                name: \'one-son-son\',\n                                                value: \'0-2-0-0-0\',\n                                                children: [\n                                                    {\n                                                        name: \'one-child-son-son\',\n                                                        value: \'0-2-0-0-0-0\',\n                                                        children: [\n                                                            {\n                                                                name: \'one-son-son-son\',\n                                                                value: \'0-2-0-0-0-0-0\',\n                                                                children: [\n                                                                    {\n                                                                        name: \'one-child-son-son-son\',\n                                                                        value: \'0-2-0-0-0-0-0-0\'\n                                                                    },\n                                                                    {\n                                                                        name: \'two-child-son-son-son\',\n                                                                        value: \'0-2-0-0-0-0-0-1\'\n                                                                    }\n                                                                ]\n                                                            },\n                                                        ]\n                                                    },\n                                                ]\n                                            },\n                                        ]\n                                    },\n                                ]\n                            }\n                        ]\n                    },\n                    {\n                        name: \'one-child\',\n                        value: \'0-3\',\n                        children: [\n                            {\n                                name: \'two-son\',\n                                value: \'0-3-0\',\n                                children: [\n                                    {\n                                        name: \'one-child-son\',\n                                        value: \'0-3-0-0\',\n                                        children: [\n                                            {\n                                                name: \'one-son-son\',\n                                                value: \'0-3-0-0-0\',\n                                                children: [\n                                                    {\n                                                        name: \'one-child-son-son\',\n                                                        value: \'0-3-0-0-0-0\',\n                                                        children: [\n                                                            {\n                                                                name: \'one-son-son-son\',\n                                                                value: \'0-3-0-0-0-0-0\',\n                                                                children: [\n                                                                    {\n                                                                        name: \'one-child-son-son-son\',\n                                                                        value: \'0-3-0-0-0-0-0-0\'\n                                                                    },\n                                                                    {\n                                                                        name: \'two-child-son-son-son\',\n                                                                        value: \'0-3-0-0-0-0-0-1\'\n                                                                    }\n                                                                ]\n                                                            },\n                                                        ]\n                                                    },\n                                                ]\n                                            },\n                                        ]\n                                    },\n                                ]\n                            }\n                        ]\n                    },\n                    {\n                        name: \'two-child\',\n                        value: \'0-4\'\n                    },\n                    {\n                        name: \'one-child\',\n                        value: \'0-5\'\n                    },\n                    {\n                        name: \'two-child\',\n                        value: \'0-6\'\n                    },\n                    {\n                        name: \'one-child\',\n                        value: \'0-1\'\n                    },\n                    {\n                        name: \'two-child\',\n                        value: \'0-2\'\n                    },\n                    {\n                        name: \'one-child\',\n                        value: \'0-1\'\n                    },\n                    {\n                        name: \'two-child\',\n                        value: \'0-2\'\n                    },\n                    {\n                        name: \'one-child\',\n                        value: \'0-1\',\n                        children: [\n                            {\n                                name: \'two-son\',\n                                value: \'1-2-0\'\n                            }\n                        ]\n                    },\n                    {\n                        name: \'two-child\',\n                        value: \'0-2\',\n                        children: [\n                            {\n                                name: \'two-son\',\n                                value: \'1-2-0\'\n                            }\n                        ]\n                    },\n                    {\n                        name: \'one-child\',\n                        value: \'0-1\',\n                        children: [\n                            {\n                                name: \'two-son\',\n                                value: \'1-2-0\'\n                            }\n                        ]\n                    },\n                    {\n                        name: \'two-child\',\n                        value: \'0-2\'\n                    }\n                ]\n            },\n            {\n                name: \'two\',\n                value: \'1\',\n                children: [\n                    {\n                        name: \'one-child\',\n                        value: \'1-1\',\n                        children: [\n                            {\n                                name: \'two-son\',\n                                value: \'1-2-0\'\n                            }\n                        ]\n                    },\n                    {\n                        name: \'two-child\',\n                        value: \'1-2\',\n                        children: [\n                            {\n                                name: \'two-son\',\n                                value: \'2-2-0\'\n                            }\n                        ]\n                    }\n                ]\n            },\n            {\n                name: \'\u7279\u65AF\u8F7C\',\n                value: \'t\'\n            }\n        ];\n        const defaultValue = [\n            \'0-3-0-0-0-0-0-0\', \'0-2-0-0-0-0-0-1\'\n        ]\n        return (\n            <Cascader data={exampleData} defaultValue={defaultValue} onChange={(v) => { console.log(\'multi change\', v); }} type="multi" placement=\'right\'>\n                <Button className="mcds-button__brand">\u591A\u9009\u62E9</Button>\n            </Cascader>;\n        );\n    }\n}';
var introMulti = '##### Cascader提供以下参数\r\n\r\n- Cascader-Multi\r\n\t- data 基本数据（name为显示的字，value为调用的值，有下一级则加children）\r\n\t- defaultValue 默认被选中（数组，传入对象的value即可，如果是多层级的，传入最小children的value）\r\n\t- placement 弹出方向 支持left/right 如left为向左弹（children也是向左）\t\r\n\t- type multi单选或者多选,多选可不填\t\r\n\t- onChange 父亲传入的回掉\r\n';
var introSingle = '##### 单选不同处\r\n\r\n- defaultValue 默认被选中 传入数字或者字符串\r\n\t- type single单选或者多选,默认为单选\r\n';

exports.default = [{
    id: 'Cascader-Multi',
    element: _react2.default.createElement(MultiCascader, null),
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: introMulti }),
    code: MultiCascaderExample
}, {
    id: 'Cascader-Single',
    element: _react2.default.createElement(_singleSelect2.default, null),
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: introSingle }),
    code: SingleCascaderExample
    // {
    //     id: 'MenuSelect-single',
    //     element: <SingleSelect />,
    //     intro: <ReactMarkdown className="markdown" source={MenuSelectIntro} />
    // },
    // {
    //     id: 'MenuSelect-multi',
    //     element: <MultiSelect />,
    //     intro: '这是一个多选',
    //     code: MultiSelectCode
    // },
    // {
    //     id: 'PopoverSelect',
    //     element: PopoverSelect,
    //     intro: <ReactMarkdown className="markdown" source={PopSelectIntro} />
    // },
    // {
    //     id: 'PopoverSelect-noarrow',
    //     element: PopoverSelectWithoutArrow,
    //     intro: ''
    // },
    // {
    //     id: 'MenuExtend',
    //     element: <ExtendSelect />,
    //     intro: <ReactMarkdown className="markdown" source={ExtendIntro} />,
    //     code: ExtendSelectCode
    // }
}];
module.exports = exports['default'];
