// carbon/lib/components/file-selector/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkdown = require('react-markdown');

var _reactMarkdown2 = _interopRequireDefault(_reactMarkdown);

var _index = require('./index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var IntegratedDefault = _react2.default.createElement(
    'div',
    { style: { width: '320px', height: '320px' } },
    _react2.default.createElement(
        _index.IntegratedFileSelector,
        null,
        '\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.'
    )
);

var IntegratedDrag = _react2.default.createElement(
    'div',
    { style: { width: '320px', height: '320px' } },
    _react2.default.createElement(
        _index.IntegratedFileSelector,
        { drag: true, dropContent: '\u6216\u8005\u62D6\u653E\u6587\u4EF6' },
        '\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.'
    )
);

var IntegratedDragover = _react2.default.createElement(
    'div',
    { style: { width: '320px', height: '320px' } },
    _react2.default.createElement(
        _index.IntegratedFileSelector,
        { drag: true, draggover: true, dropContent: '\u6216\u8005\u62D6\u653E\u6587\u4EF6' },
        '\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.'
    )
);

var IntegratedDragoverError = _react2.default.createElement(
    'div',
    { style: { width: '320px', height: '320px' } },
    _react2.default.createElement(
        _index.IntegratedFileSelector,
        { drag: true, draggoverError: true, errorMsg: '\u9519\u8BEF\u4FE1\u606F\u5199\u5728\u8FD9\u91CC' },
        '\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.\u8FD9\u662Ffile-selector.'
    )
);

exports.default = [{
    id: 'FilesSel',
    element: _react2.default.createElement(_index.FileSelector, {
        label: 'Account Name',
        drop: true,
        Icon: 'mcds-icon__upload-line-20',
        error: 'File type not supported',
        accept: 'image/png',
        iconContent: '\u4E0A\u4F20\u6570\u636E\u6587\u4EF6',
        dropContent: '\u6216\u62D6\u653E\u6570\u636E\u6587\u4EF6\u5230\u8FD9\u91CC' }),
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: '#### FileSelector\r\n* 支持传入 label \r\n* drop 是否支持拖拽\r\n* Icon 图标\r\n* error 错误信息\r\n* accept 支持上传的格式  默认为\"image/png\"\r\n* iconContent icon右面的文字\r\n* dropContent 拖放数据文件到这里' })
}, {
    id: 'NoBorderFiles',
    element: _react2.default.createElement(_index.FileSelector, { label: 'Account Name', Icon: 'mcds-icon__upload-line-20', error: 'File type not supported', accept: 'image/png', iconContent: '\u4E0A\u4F20\u6570\u636E\u6587\u4EF6' })
}, {
    id: 'Image',
    element: _react2.default.createElement(_index.FileSelector, {
        drop: true,
        className: 'mcds-file-selector__images mcds-data__import',
        label: 'Account Name',
        Icon: 'mcds-icon__upload-line-20',
        error: 'File type not supported',
        accept: 'image/png',
        iconContent: '\u4E0A\u4F20\u6570\u636E\u6587\u4EF6',
        dropContent: '\u6216\u62D6\u653E\u6570\u636E\u6587\u4EF6\u5230\u8FD9\u91CC' })
}, {
    id: 'IntegratedDefault',
    element: IntegratedDefault,
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: '#### FileSelector\r\n* 支持传入 label \r\n* drop 是否支持拖拽\r\n* draggover 值为true/false 用来判断是否添加文件拖拽到固定区域的样式\r\n* draggoverError 为true时 errorMsg生效  值为:错误信息\r\n* accept 支持上传的格式  默认为\"image/png\"\r\n* dropContent 拖放数据文件到这里' })
}, {
    id: 'IntegratedDrag',
    element: IntegratedDrag
}, {
    id: 'IntegratedDragover',
    element: IntegratedDragover
}, {
    id: 'IntegratedDragoverError',
    element: IntegratedDragoverError
}];
module.exports = exports['default'];
