// carbon/lib/components/notification/flavors/alert/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _notification = require('../base/notification');

var _notification2 = _interopRequireDefault(_notification);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Alert = function (_Component) {
    (0, _inherits3.default)(Alert, _Component);

    function Alert(props) {
        (0, _classCallCheck3.default)(this, Alert);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Alert.__proto__ || (0, _getPrototypeOf2.default)(Alert)).call(this, props));

        _this.state = {
            message: '',
            isActive: false
        };
        _this.show = _this.show.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(Alert, [{
        key: 'show',
        value: function show(options) {
            var onDismiss = this.onDismiss.bind(this);
            this.setState((0, _extends3.default)({}, options, {
                isActive: true,
                onDismiss: onDismiss
            }));
        }
    }, {
        key: 'onDismiss',
        value: function onDismiss() {
            this.setState({
                isActive: false
            });
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(_notification2.default, this.state);
        }
    }]);
    return Alert;
}(_react.Component);

exports.default = Alert;
module.exports = exports['default'];
