// carbon/lib/components/editor/flavors/base/src/plugins/buttons/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.ImageButton = exports.LinkButton = exports.AlignCenterButton = exports.AlignRightButton = exports.AlignLeftButton = exports.UnorderedListButton = exports.OrderedListButton = exports.UnderlineButton = exports.ItalicButton = exports.BoldButton = exports.FontSizeButton = exports.ColorButton = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _components = require('../../components');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var StyleComponent = function (_Component) {
    (0, _inherits3.default)(StyleComponent, _Component);

    function StyleComponent() {
        (0, _classCallCheck3.default)(this, StyleComponent);
        return (0, _possibleConstructorReturn3.default)(this, (StyleComponent.__proto__ || (0, _getPrototypeOf2.default)(StyleComponent)).apply(this, arguments));
    }

    (0, _createClass3.default)(StyleComponent, [{
        key: 'render',
        value: function render() {
            var _props = this.props,
                onClick = _props.onClick,
                type = _props.type,
                props = (0, _objectWithoutProperties3.default)(_props, ['onClick', 'type']);

            return _react2.default.createElement(_components.IconButton, (0, _extends3.default)({}, props, {
                onClick: onClick,
                name: type }));
        }
    }]);
    return StyleComponent;
}(_react.Component); /* eslint-disable */


StyleComponent.propTypes = {
    onClick: _propTypes2.default.func,
    type: _propTypes2.default.string
};
var Wrap = function Wrap(WrapedComponent) {
    return function (type) {
        var WrapComponent = function (_Component2) {
            (0, _inherits3.default)(WrapComponent, _Component2);

            function WrapComponent() {
                (0, _classCallCheck3.default)(this, WrapComponent);
                return (0, _possibleConstructorReturn3.default)(this, (WrapComponent.__proto__ || (0, _getPrototypeOf2.default)(WrapComponent)).apply(this, arguments));
            }

            (0, _createClass3.default)(WrapComponent, [{
                key: 'render',
                value: function render() {
                    return _react2.default.createElement(WrapedComponent, (0, _extends3.default)({}, this.props, {
                        type: type
                    }));
                }
            }]);
            return WrapComponent;
        }(_react.Component);

        return WrapComponent;
    };
};

var StyleComponentWrap = Wrap(StyleComponent);

var types = {
    color: 'font-color-18',
    fontSize: 'font-scale-18',
    bold: 'bold-18',
    italic: 'italic-18',
    underline: 'underline-18',
    orderedList: 'ordered-list-18',
    unorderedList: 'unordered-list-18',
    alignLeft: 'align-left-18',
    alignCenter: 'center-18',
    alignRight: 'align-right-18',
    link: 'insert-link-18',
    image: 'insert-image-18'
};
var PREFIX = 'mcds-icon';

var ColorButton = StyleComponentWrap(PREFIX + '__' + types.color);
var FontSizeButton = StyleComponentWrap(PREFIX + '__' + types.fontSize);
var BoldButton = StyleComponentWrap(PREFIX + '__' + types.bold);
var ItalicButton = StyleComponentWrap(PREFIX + '__' + types.italic);
var UnderlineButton = StyleComponentWrap(PREFIX + '__' + types.underline);
var OrderedListButton = StyleComponentWrap(PREFIX + '__' + types.orderedList);
var UnorderedListButton = StyleComponentWrap(PREFIX + '__' + types.unorderedList);
var AlignLeftButton = StyleComponentWrap(PREFIX + '__' + types.alignLeft);
var AlignCenterButton = StyleComponentWrap(PREFIX + '__' + types.alignCenter);
var AlignRightButton = StyleComponentWrap(PREFIX + '__' + types.alignRight);
var LinkButton = StyleComponentWrap(PREFIX + '__' + types.link);
var ImageButton = StyleComponentWrap(PREFIX + '__' + types.image);

exports.ColorButton = ColorButton;
exports.FontSizeButton = FontSizeButton;
exports.BoldButton = BoldButton;
exports.ItalicButton = ItalicButton;
exports.UnderlineButton = UnderlineButton;
exports.OrderedListButton = OrderedListButton;
exports.UnorderedListButton = UnorderedListButton;
exports.AlignLeftButton = AlignLeftButton;
exports.AlignRightButton = AlignRightButton;
exports.AlignCenterButton = AlignCenterButton;
exports.LinkButton = LinkButton;
exports.ImageButton = ImageButton;
