// carbon/lib/components/form/flavors/checkbox-group-alertnate/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AlternateCheckbox = function (_React$Component) {
    (0, _inherits3.default)(AlternateCheckbox, _React$Component);

    function AlternateCheckbox(props) {
        (0, _classCallCheck3.default)(this, AlternateCheckbox);

        var _this = (0, _possibleConstructorReturn3.default)(this, (AlternateCheckbox.__proto__ || (0, _getPrototypeOf2.default)(AlternateCheckbox)).call(this, props));

        _this.state = {
            checked: !!_this.props.checked
        };
        return _this;
    }

    (0, _createClass3.default)(AlternateCheckbox, [{
        key: 'handleChange',
        value: function handleChange(e) {
            this.setState({ checked: e.target.checked });
            this.props.callback(e.target.checked);
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'span',
                { className: (0, _classnames2.default)('mcds-checkbox__button', this.props.className) },
                _react2.default.createElement('input', {
                    name: this.props.name,
                    type: 'checkbox',
                    id: this.props.id,
                    ref: this.props.id,
                    value: this.props.value,
                    disabled: this.props.disabled,
                    checked: this.state.checked,
                    onChange: this.handleChange.bind(this) }),
                _react2.default.createElement(
                    'label',
                    { className: 'mcds-checkbox-button__label', htmlFor: this.props.id },
                    _react2.default.createElement(
                        'span',
                        { className: 'mcds-checkbox__faux' },
                        this.props.children
                    )
                )
            );
        }
    }]);
    return AlternateCheckbox;
}(_react2.default.Component);

AlternateCheckbox.defaultProps = {
    callback: function callback(val) {
        return val;
    }
};


AlternateCheckbox.propTypes = {
    id: _propTypes2.default.string.isRequired,
    name: _propTypes2.default.string,
    value: _propTypes2.default.string.isRequired,
    className: _propTypes2.default.string,
    disabled: _propTypes2.default.string,
    checked: _propTypes2.default.string,
    children: _propTypes2.default.any,
    callback: _propTypes2.default.func
};

var AlternateCheckboxGroup = function AlternateCheckboxGroup(_ref) {
    var className = _ref.className,
        label = _ref.label,
        children = _ref.children;
    return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)(className) },
        label ? _react2.default.createElement(
            'div',
            { className: 'mcds-form-element__label' },
            label
        ) : null,
        _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(
                'div',
                { className: 'mcds-checkbox-button__group' },
                children
            )
        )
    );
};

AlternateCheckboxGroup.propTypes = {
    className: _propTypes2.default.string,
    label: _propTypes2.default.string,
    children: _propTypes2.default.any
};

exports.default = {
    AlternateCheckbox: AlternateCheckbox,
    AlternateCheckboxGroup: AlternateCheckboxGroup
};
module.exports = exports['default'];
