// carbon/lib/components/form/flavors/checkbox/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Checkbox = function (_React$Component) {
    (0, _inherits3.default)(Checkbox, _React$Component);

    function Checkbox(props) {
        (0, _classCallCheck3.default)(this, Checkbox);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Checkbox.__proto__ || (0, _getPrototypeOf2.default)(Checkbox)).call(this, props));

        _this.state = {
            checked: !!_this.props.checked
        };
        return _this;
    }

    (0, _createClass3.default)(Checkbox, [{
        key: 'componentWillReceiveProps',
        value: function componentWillReceiveProps(nextProps) {
            this.state = {
                checked: !!nextProps.checked
            };
        }
    }, {
        key: 'handleChange',
        value: function handleChange(e) {
            this.setState({ checked: e.target.checked });
            this.props.onChange(e.target.checked);
        }
    }, {
        key: 'handleClick',
        value: function handleClick() {
            this.setState({
                checked: !this.state.checked
            });
        }
    }, {
        key: '_renderCheckbox',
        value: function _renderCheckbox() {
            var checked = this.state.checked;
            var className = checked ? 'mcds-checkbox mcds-checkbox__checked' : 'mcds-checkbox';
            return _react2.default.createElement(
                'div',
                { className: this.props.className },
                _react2.default.createElement(
                    'label',
                    { className: 'mcds-checkbox__wrapper ' },
                    _react2.default.createElement(
                        'span',
                        { className: className },
                        _react2.default.createElement('input', {
                            className: 'mcds-checkbox__input',
                            type: 'checkbox',
                            id: this.props.id,
                            name: this.props.name,
                            checked: checked,
                            disabled: this.props.disabled,
                            onChange: this.handleChange.bind(this),
                            ref: 'checkbox',
                            onClick: this.handleClick.bind(this) }),
                        _react2.default.createElement('span', { className: 'mcds-checkbox__inner' })
                    ),
                    this.props.label ? _react2.default.createElement(
                        'span',
                        null,
                        this.props.label
                    ) : null
                )
            );
        }
    }, {
        key: '_renderRequiredCheckbox',
        value: function _renderRequiredCheckbox() {
            var checked = this.state.checked;
            var className = checked ? 'mcds-checkbox mcds-checkbox__checked' : 'mcds-checkbox';
            return _react2.default.createElement(
                'div',
                { className: this.props.className },
                _react2.default.createElement(
                    'label',
                    { className: 'mcds-checkbox__wrapper ' },
                    _react2.default.createElement(
                        'span',
                        { className: className },
                        _react2.default.createElement('input', {
                            className: 'mcds-checkbox__input',
                            type: 'checkbox',
                            id: this.props.id,
                            name: this.props.name,
                            checked: checked,
                            disabled: this.props.disabled,
                            onChange: this.handleChange.bind(this),
                            ref: 'checkbox',
                            onClick: this.handleClick.bind(this) }),
                        _react2.default.createElement('span', { className: 'mcds-checkbox__inner' })
                    ),
                    this.props.label ? _react2.default.createElement(
                        'span',
                        null,
                        this.props.label
                    ) : null
                ),
                this.props.errorMsg ? _react2.default.createElement(
                    'span',
                    { className: 'mcds-span__required' },
                    this.props.errorMsg
                ) : null
            );
        }
    }, {
        key: '_renderIndeterminateCheckbox',
        value: function _renderIndeterminateCheckbox() {
            var checked = this.state.checked;
            var className = checked ? 'mcds-checkbox mcds-checkbox__indeterminate' : 'mcds-checkbox';
            return _react2.default.createElement(
                'div',
                { className: this.props.className },
                _react2.default.createElement(
                    'label',
                    { className: 'mcds-checkbox__wrapper' },
                    _react2.default.createElement(
                        'span',
                        { className: className },
                        _react2.default.createElement('input', {
                            className: 'mcds-checkbox__input',
                            type: 'checkbox',
                            id: this.props.id,
                            name: this.props.name,
                            checked: checked,
                            disabled: this.props.disabled,
                            onChange: this.handleChange.bind(this),
                            ref: 'checkbox',
                            onClick: this.handleClick.bind(this) }),
                        _react2.default.createElement('span', { className: 'mcds-checkbox__inner' })
                    ),
                    this.props.label ? _react2.default.createElement(
                        'span',
                        null,
                        this.props.label
                    ) : null
                )
            );
        }
    }, {
        key: 'render',
        value: function render() {

            if (this.props.indeterminate) {
                return this._renderIndeterminateCheckbox();
            }
            if (this.props.required) {
                return this._renderRequiredCheckbox();
            }
            return this._renderCheckbox();
        }
    }]);
    return Checkbox;
}(_react2.default.Component);

Checkbox.defaultProps = {
    handleClick: function handleClick() {},
    onChange: function onChange() {}
};

Checkbox.propTypes = {
    id: _propTypes2.default.string,
    name: _propTypes2.default.string,
    label: _propTypes2.default.string,
    errorMsg: _propTypes2.default.string,
    className: _propTypes2.default.string,
    checked: _propTypes2.default.bool,
    indeterminate: _propTypes2.default.bool,
    disabled: _propTypes2.default.bool,
    required: _propTypes2.default.bool,
    handleClick: _propTypes2.default.func,
    onChange: _propTypes2.default.func
};

var CheckboxGroup = function CheckboxGroup(_ref) {
    var groupLable = _ref.groupLable,
        className = _ref.className,
        children = _ref.children;
    return _react2.default.createElement(
        'div',
        { className: className },
        _react2.default.createElement(
            'span',
            { className: 'mcds-ckeckbox-group__label' },
            groupLable
        ),
        _react2.default.createElement(
            'div',
            null,
            children
        )
    );
};

CheckboxGroup.propTypes = {
    groupLable: _propTypes2.default.string,
    children: _propTypes2.default.any,
    className: _propTypes2.default.string
};

var RequiredCheckboxGroup = function RequiredCheckboxGroup(_ref2) {
    var groupLable = _ref2.groupLable,
        children = _ref2.children,
        errorMsg = _ref2.errorMsg,
        className = _ref2.className;
    return _react2.default.createElement(
        'div',
        { className: className },
        _react2.default.createElement(
            'span',
            { className: 'mcds-ckeckbox-group__label' },
            _react2.default.createElement(
                'span',
                { className: 'mcds-span__required' },
                '*'
            ),
            groupLable
        ),
        _react2.default.createElement(
            'div',
            null,
            children,
            _react2.default.createElement(
                'span',
                { className: 'mcds-span__required' },
                errorMsg
            )
        )
    );
};

RequiredCheckboxGroup.propTypes = {
    groupLable: _propTypes2.default.string,
    errorMsg: _propTypes2.default.string,
    children: _propTypes2.default.any,
    className: _propTypes2.default.string
};

exports.default = {
    Checkbox: Checkbox,
    CheckboxGroup: CheckboxGroup,
    RequiredCheckboxGroup: RequiredCheckboxGroup
};
module.exports = exports['default'];
