// carbon/lib/components/popover/flavor/_base/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Trigger = exports.Popover = exports.PopoverBody = exports.PopoverHead = exports.getPosition = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _dom = require('../../../../utils/dom');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * [获取目标节点在dom中的位置]
 * @param  {[reactdom/dom]} target    [react节点.或者dom节点]
 * @param  {[direction]} direction [暂时无用]
 * @return {[object]}           [{top,left,width,height}}]
 */
/**
 * 这是一个popover的基础,所有的popover都是基于base文件衍生出来的
 * base 提供显示功能,不做逻辑功能,不对外开放.
 * props{
 *     direction:{ //方向参数
 *         top:{
 *             left,
 *             right,
 *         }
 *         bottom:{
 *             left,
 *             right;
 *         }
 *         left:{
 *             top,
 *             bottom,
 *         }
 *         right:{
 *             top,
 *             bottom
 *         }
 *     }
 * }
 */
var getPosition = exports.getPosition = function getPosition(target) {
    // console.log(typeof target,'target',ReactDOM.findDOMNode(target));
    if (!target) {
        return null;
    }
    var _target = target;
    return (0, _dom.getDomOffset)(_target);
};
/**
 * popover的关部样式
 * @param  {[type]} options.children  [description]
 * @param  {[type]} options.className [description]
 * @return {[type]}                   [description]
 */
var PopoverHead = function PopoverHead(_ref) {
    var children = _ref.children,
        className = _ref.className,
        others = (0, _objectWithoutProperties3.default)(_ref, ['children', 'className']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-popover__header', className) }),
        children
    );
};

exports.PopoverHead = PopoverHead;
PopoverHead.propTypes = {
    children: _propTypes2.default.element.isRequired,
    className: _propTypes2.default.string
};

var PopoverBody = function PopoverBody(_ref2) {
    var className = _ref2.className,
        children = _ref2.children,
        others = (0, _objectWithoutProperties3.default)(_ref2, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-popover__body', className) }),
        children
    );
};

exports.PopoverBody = PopoverBody;
PopoverBody.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string
};

var Popover = function Popover(_ref3) {
    var theme = _ref3.theme,
        className = _ref3.className,
        type = _ref3.type,
        children = _ref3.children,
        others = (0, _objectWithoutProperties3.default)(_ref3, ['theme', 'className', 'type', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { 'data-type': type, className: (0, _classnames2.default)('mcds-popover mcds-popover__' + theme, className) }),
        children
    );
};
exports.Popover = Popover;
Popover.defaultProps = {
    theme: 'default',
    type: 'popover'
};
Popover.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string,
    theme: _propTypes2.default.oneOf(['default', 'info', 'error', 'warning', 'success']),
    type: _propTypes2.default.oneOf(['tooltip', 'panel', 'popover'])
};

var Trigger = function (_React$Component) {
    (0, _inherits3.default)(Trigger, _React$Component);

    function Trigger() {
        (0, _classCallCheck3.default)(this, Trigger);
        return (0, _possibleConstructorReturn3.default)(this, (Trigger.__proto__ || (0, _getPrototypeOf2.default)(Trigger)).apply(this, arguments));
    }

    (0, _createClass3.default)(Trigger, [{
        key: 'render',
        value: function render() {
            var _this2 = this;

            var _props = this.props,
                className = _props.className,
                children = _props.children,
                others = (0, _objectWithoutProperties3.default)(_props, ['className', 'children']);

            return _react2.default.createElement(
                'div',
                (0, _extends3.default)({}, others, { ref: function ref(node) {
                        return _this2.node = node;
                    }, className: (0, _classnames2.default)('mcds-popover-trigger mcds-popover-trigger__open', className) }),
                children
            );
        }
    }]);
    return Trigger;
}(_react2.default.Component);

exports.Trigger = Trigger;
Trigger.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string
};
