// carbon/lib/components/editor/flavors/base/src/components/popover/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

var _clickOutside = require('../click-outside');

var _clickOutside2 = _interopRequireDefault(_clickOutside);

var _utils = require('../../utils');

var _util = require('./util');

var _util2 = _interopRequireDefault(_util);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// const DEFAULT_TOP = -10000;
// const DEFAULT_LEFT = -10000;

/* eslint-disable */
/**
 * 气泡卡片
 * const content = (
 * 	<div> content1 </div>
 * 	<div> content2 </div>
 * )
 * <Popover content={content} position="bottom">
 * 	<button>点击弹出<button>
 * </Popover>
 */

var Popover = function (_Component) {
    (0, _inherits3.default)(Popover, _Component);

    function Popover() {
        (0, _classCallCheck3.default)(this, Popover);
        return (0, _possibleConstructorReturn3.default)(this, (Popover.__proto__ || (0, _getPrototypeOf2.default)(Popover)).apply(this, arguments));
    }

    (0, _createClass3.default)(Popover, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            var _this2 = this;

            // render to body
            this.mountNode = document.createElement('div');
            this.mountNode.style.position = 'absolute';
            this.mountNode.style.zIndex = 9999;
            var content = _react2.default.createElement(
                _clickOutside2.default,
                { onClickOutside: function onClickOutside() {
                        return _this2.hide();
                    } },
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-editor__popover' },
                    (0, _react.cloneElement)(this.props.content)
                )
            );
            _reactDom2.default.unstable_renderSubtreeIntoContainer(this, content, this.mountNode);
            this.hide();
            document.body.appendChild(this.mountNode);
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            _reactDom2.default.unmountComponentAtNode(this.mountNode);
            document.body.removeChild(this.mountNode);
            this.mountNode = null;
        }
    }, {
        key: 'setPosition',
        value: function setPosition() {
            var target = _reactDom2.default.findDOMNode(this);
            var _props = this.props,
                placement = _props.placement,
                position = _props.position;

            var eleOffset = (0, _utils.getOffset)(this.mountNode);
            var offset = (0, _utils.getOffset)(target);
            if (offset && eleOffset) {
                var _getRelativePosition = (0, _util2.default)(offset, eleOffset, placement),
                    top = _getRelativePosition.top,
                    left = _getRelativePosition.left;

                this.mountNode.style.top = top + (position && position.top || 0) + 'px';
                this.mountNode.style.left = left + (position && position.left || 0) + 'px';
            }
        }
    }, {
        key: 'hide',
        value: function hide() {
            (0, _utils.hide)(this.mountNode);
            if (this.props.onHide) {
                this.props.onHide();
            }
        }
    }, {
        key: 'show',
        value: function show() {
            (0, _utils.show)(this.mountNode);
            this.setPosition();
            if (this.props.onShow) {
                this.props.onShow();
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var trigger = _react2.default.Children.only(this.props.children);
            var props = {
                onClick: (0, _utils.createChainedFunction)(trigger.props.onClick, this.show.bind(this))
            };
            return (0, _react.cloneElement)(trigger, props);
        }
    }]);
    return Popover;
}(_react.Component);

exports.default = Popover;


Popover.propTypes = {
    children: _propTypes2.default.node,
    content: _propTypes2.default.node,
    style: _propTypes2.default.object,
    placement: _propTypes2.default.string,
    position: _propTypes2.default.object,
    onShow: _propTypes2.default.func,
    onHide: _propTypes2.default.func
};

Popover.defaultProps = {
    placement: 'bottom',
    style: {}
};
module.exports = exports['default'];
