// carbon/lib/components/editor/flavors/base/src/plugins/blocks/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.AlignRightComponent = exports.AlignCenterComponent = exports.AlignLeftComponent = exports.UnorderedListComponent = exports.OrderedListComponent = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _draftJs = require('draft-js');

var _buttons = require('../buttons');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Wrap = function Wrap(WrapedComponent) {
    return function (type) {
        var WrapComponent = function (_Component) {
            (0, _inherits3.default)(WrapComponent, _Component);

            function WrapComponent() {
                (0, _classCallCheck3.default)(this, WrapComponent);
                return (0, _possibleConstructorReturn3.default)(this, (WrapComponent.__proto__ || (0, _getPrototypeOf2.default)(WrapComponent)).apply(this, arguments));
            }

            (0, _createClass3.default)(WrapComponent, [{
                key: 'handleChange',
                value: function handleChange() {
                    var _props = this.props,
                        editorState = _props.editorState,
                        onChange = _props.onChange;

                    onChange(_draftJs.RichUtils.toggleBlockType(editorState, type));
                }
            }, {
                key: 'render',
                value: function render() {
                    var editorState = this.props.editorState;
                    var selection = editorState.getSelection();
                    var blockType = editorState.getCurrentContent().getBlockForKey(selection.getStartKey()).getType();
                    return _react2.default.createElement(WrapedComponent, (0, _extends3.default)({}, this.props, {
                        active: blockType === type,
                        onClick: this.handleChange.bind(this) }));
                }
            }]);
            return WrapComponent;
        }(_react.Component);

        WrapComponent.propTypes = {
            editorState: _propTypes2.default.object,
            onChange: _propTypes2.default.func
        };
        return WrapComponent;
    };
};

var OrderedListComponent = Wrap(_buttons.OrderedListButton)('ordered-list-item');
var UnorderedListComponent = Wrap(_buttons.UnorderedListButton)('unordered-list-item');
var AlignLeftComponent = Wrap(_buttons.AlignLeftButton)('header-four');
var AlignCenterComponent = Wrap(_buttons.AlignCenterButton)('header-five');
var AlignRightComponent = Wrap(_buttons.AlignRightButton)('header-six');

exports.OrderedListComponent = OrderedListComponent;
exports.UnorderedListComponent = UnorderedListComponent;
exports.AlignLeftComponent = AlignLeftComponent;
exports.AlignCenterComponent = AlignCenterComponent;
exports.AlignRightComponent = AlignRightComponent;
