// carbon/lib/components/form/flavors/combox-select/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _from = require('babel-runtime/core-js/array/from');

var _from2 = _interopRequireDefault(_from);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _dropdown = require('../../../dropdown');

var _index = require('../input/index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ComboxSelect = function (_React$Component) {
    (0, _inherits3.default)(ComboxSelect, _React$Component);

    function ComboxSelect(props) {
        (0, _classCallCheck3.default)(this, ComboxSelect);

        // 这里干掉buildDate是考虑到效率问题,每次build挺慢的
        var _this = (0, _possibleConstructorReturn3.default)(this, (ComboxSelect.__proto__ || (0, _getPrototypeOf2.default)(ComboxSelect)).call(this, props));

        _this.state = {
            // 不要在这里加入data.更新问题会很多
            display: 'none',
            defaultValue: '',
            value: typeof _this.props.value !== 'undefined' ? _this.props.value : '-1',
            searchData: _this.props.children,
            open: false
        };
        // 用来获取编辑的值.
        _this.getValue = _this.getValue.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(ComboxSelect, [{
        key: 'componentWillReceiveProps',
        value: function componentWillReceiveProps(nextProps) {
            // 干掉的原因,是为了select 显示的内容并非是option显示的内容,
            var defaultValue = typeof nextProps.value !== 'undefined' ? nextProps.value : '请选择';
            // if (defaultValue == this.state.defaultValue) {
            this.setState({ defaultValue: defaultValue });
            // }
        }
    }, {
        key: 'getValue',
        value: function getValue() {
            var _state = this.state,
                value = _state.value,
                defaultValue = _state.defaultValue;

            var result = {};
            result.value = value;
            result.text = defaultValue;
            return result;
        }
        /**
        * 用来计算显示出来的文字 .
        * if defaultValue in children
        * value = child
        * else
        * value = props.defaultValue
        * @param  {[type]} value [description]
        * @return {[type]}       [description]
        */

    }, {
        key: 'getDefaultValue',
        value: function getDefaultValue(defaultValue) {
            var child = void 0;
            _react2.default.Children.forEach(this.props.children, function (v) {
                if (v.props.value === defaultValue) {
                    child = v.props.children;
                }
            });
            if (!child) {
                child = defaultValue;
            }
            return child;
        }
    }, {
        key: '_handleOnChick',
        value: function _handleOnChick(obj) {
            this.setState({
                defaultValue: obj.props.children,
                value: obj.props.value || obj.props.children
            });
            this.refs.select.close();
            this.props.onChange(obj.props.value || obj.props.children);
        }
    }, {
        key: 'buildOptions',
        value: function buildOptions(curVal) {
            var _this2 = this;

            var options = [];
            var children = (0, _from2.default)(this.state.searchData);
            if (curVal) {
                children.length = 0;
                _react2.default.Children.forEach(this.props.children, function (v) {
                    var reg = new RegExp(curVal, 'ig');
                    if (reg.test(v.props.children)) {
                        children.push(v);
                    }
                });
            } else {
                children = this.props.children;
            }
            _react2.default.Children.forEach(children, function (v, index) {
                var isSelected = v.props.value === _this2.state.value;
                if (typeof v.props.value === 'undefined') {
                    // 如果没有传入value的话.用children去做判断
                    isSelected = String(v.props.children) === String(_this2.state.value);
                }
                var tmp = _react2.default.createElement(
                    _dropdown.DropDownItem,
                    { isSelected: isSelected, onClick: _this2._handleOnChick.bind(_this2, v), key: index },
                    v.props.children
                );
                options.push(tmp);
            });

            var content = null;
            if (options.length === 0) {
                content = _react2.default.createElement(
                    'span',
                    { className: 'mcds-select__not-found' },
                    'Not Found'
                );
            } else {
                content = _react2.default.createElement(
                    _dropdown.DropDownList,
                    null,
                    options
                );
            }

            return _react2.default.createElement(
                _dropdown.DropDown,
                { className: (0, _classnames2.default)('mcds-select__options') },
                content
            );
        }
    }, {
        key: 'handleChange',
        value: function handleChange(val) {
            if (this.state.open) {
                this.setState({
                    defaultValue: val,
                    value: val
                });
            } else {
                this.refs.select.open();
                this.setState({
                    defaultValue: val,
                    value: val
                });
            }
        }
    }, {
        key: 'renderDisable',
        value: function renderDisable() {
            var _props$disabled = this.props.disabled,
                disabled = _props$disabled === undefined ? false : _props$disabled;

            if (disabled) {
                return _react2.default.createElement(_index.Input, { disabled: true, className: 'mcds-select__disabled', iconRight: _react2.default.createElement('span', { className: 'mcds-icon__updown-triangles-solid-14' }), placeholder: this.props.placeholder });
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var _this3 = this;

            var options = this.buildOptions(this.state.defaultValue);
            var _props = this.props,
                disabled = _props.disabled,
                error = _props.error;

            var requiredSpan = this.props.required ? _react2.default.createElement(
                'span',
                { className: 'mcds-span__required' },
                '*'
            ) : null;
            var selectLabel = this.props.label ? _react2.default.createElement(
                'label',
                { className: 'mcds-label', htmlFor: this.props.id },
                this.props.label
            ) : null;
            return _react2.default.createElement(
                'div',
                { className: (0, _classnames2.default)('mcds-element__container', this.props.className, { 'mcds-input__border': error }), ref: function ref(node) {
                        _this3.node = node;
                    } },
                requiredSpan,
                ' ',
                selectLabel,
                disabled ? this.renderDisable() : _react2.default.createElement(
                    _dropdown.DropDownTrigger,
                    {
                        onOpen: function onOpen() {
                            _this3.setState({
                                open: true
                            });
                        },
                        onClose: function onClose() {
                            _this3.setState({
                                open: false
                            });
                        },
                        target: 'body',
                        synchWidth: true,
                        ref: 'select',
                        className: 'mcds-select__control' },
                    _react2.default.createElement(_index.Input, { iconRight: _react2.default.createElement('span', { className: 'mcds-icon__updown-triangles-solid-14' }), placeholder: this.props.placeholder, value: this.state.defaultValue, onChange: this.handleChange.bind(this) }),
                    options
                )
            );
        }
    }]);
    return ComboxSelect;
}(_react2.default.Component); /**
                               * 相当于select组件的衍生版.样式存放在select/css
                               */


ComboxSelect.propTypes = {
    id: _propTypes2.default.string,
    value: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.number]),
    label: _propTypes2.default.string,
    className: _propTypes2.default.string,
    placeholder: _propTypes2.default.string,
    children: _propTypes2.default.any,
    disabled: _propTypes2.default.bool,
    onChange: _propTypes2.default.func,
    required: _propTypes2.default.bool,
    error: _propTypes2.default.bool
};

ComboxSelect.defaultProps = {
    onChange: function onChange() {},
    placeHolder: '请选择'
};

exports.default = ComboxSelect;
module.exports = exports['default'];
