// carbon/lib/components/notification/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkdown = require('react-markdown');

var _reactMarkdown2 = _interopRequireDefault(_reactMarkdown);

var _index = require('../index');

var _index2 = require('./index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ExampleCode = '\nimport Notifications, {notify} from \'***\';\n...\nclass Example extends Component{\n    constructor(props) {\n        super(props);\n    }\n    add() {\n        notify.add({\n            message: \'Notification\',\n            theme: \'info\',\n            dismissAfter: 3000\n        });\n    }\n    show() {\n        notify.show(\'12313\');\n    }\n    render() {\n        return (\n            <div>\n                <div>\n                    <p>\u8FD9\u662F\u4E00\u4E2A\u961F\u5217\u63D0\u793A\u4FE1\u606F.</p>\n                    <Button\n                        className="mcds-button__neutral"\n                        onClick={this.add.bind(this)}>toast</Button>\n                </div>\n\n                <div>\n                    <p>\u8FD9\u662F\u5355\u6311\u7684\u63D0\u793A\u4FE1\u606F.</p>\n                    <Button className="mcds-button__neutral" onClick={this.show.bind(this)}>alert</Button>\n                </div>\n                //Notifications \u58F0\u660E\u4E00\u6B21\u5C31\u597D,\u6700\u597D\u5728\u5916\u5C42\u7EC4\u4EF6\u4E2D\u8C03\u7528 \n                <Notifications />\n            </div>\n        );\n    }\n}\n';

var Example = function (_Component) {
    (0, _inherits3.default)(Example, _Component);

    function Example(props) {
        (0, _classCallCheck3.default)(this, Example);
        return (0, _possibleConstructorReturn3.default)(this, (Example.__proto__ || (0, _getPrototypeOf2.default)(Example)).call(this, props));
    }

    (0, _createClass3.default)(Example, [{
        key: 'add',
        value: function add() {
            _index2.notify.add({
                message: 'Notification',
                theme: 'info',
                dismissAfter: 3000
            });
        }
    }, {
        key: 'show',
        value: function show() {
            _index2.notify.show('12313');
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(
                    'div',
                    null,
                    _react2.default.createElement(
                        'p',
                        null,
                        '\u8FD9\u662F\u4E00\u4E2A\u961F\u5217\u63D0\u793A\u4FE1\u606F.'
                    ),
                    _react2.default.createElement(
                        _index.Button,
                        {
                            className: 'mcds-button__neutral',
                            onClick: this.add.bind(this) },
                        'toast'
                    )
                ),
                _react2.default.createElement(
                    'div',
                    null,
                    _react2.default.createElement(
                        'p',
                        null,
                        '\u8FD9\u662F\u5355\u6311\u7684\u63D0\u793A\u4FE1\u606F.'
                    ),
                    _react2.default.createElement(
                        _index.Button,
                        { className: 'mcds-button__neutral', onClick: this.show.bind(this) },
                        'alert'
                    )
                ),
                _react2.default.createElement(_index2.Notifications, null)
            );
        }
    }]);
    return Example;
}(_react.Component);

var intro = "##### Notifications有二种表现形式\r\n\r\n- alert\r\n- toast\r\n\r\n###### alert是长期提醒,必须手动才可以关闭 调用方式 \r\n\r\n```\r\nimport {notify} from 'somewhere';\r\nnotify.show('something');\r\n```\r\n或者\r\n\r\n```\r\nwindow.notify.show('abc');\r\n或者\r\nwindow.notify.show({message:'abc',theme,dismissAfter})\r\n```\r\n\r\n###### toast是自动显示和隐藏的\r\n```\r\nimport {notify} from 'somewhere';\r\nnotify.add('something');\r\n```\r\n或者\r\n\r\n```\r\nwindow.notify.add('abc');\r\n或者\r\nwindow.notify.add({message:'abc',theme,dismissAfter})\r\n```\r\n\r\n##### options常用属性\r\n\r\n- message 提示的内容\r\n- theme 主题\r\n\t- info\r\n\t- error \r\n\t- base\r\n\t- warning\r\n\t- success\r\n- dismissAfter  多少毫秒后消失 如果为false默认为不消失\r\n- closed 是否显示右侧关闭图标\r\n- activeClassName\t显示后的样式,可以自定义\r\n";
exports.default = [{
    id: 'Notifications',
    element: _react2.default.createElement(
        Example,
        null,
        '123'
    ),
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: intro }),
    code: ExampleCode
}];
module.exports = exports['default'];
