// carbon/lib/components/editor/flavors/base/src/plugins/inlines/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.UnderlineComponent = exports.ItalicComponent = exports.BoldComponent = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _draftJs = require('draft-js');

var _buttons = require('../buttons');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Wrap = function Wrap(WrapedComponent) {
    return function (style) {
        var WrapComponent = function (_Component) {
            (0, _inherits3.default)(WrapComponent, _Component);

            function WrapComponent() {
                (0, _classCallCheck3.default)(this, WrapComponent);
                return (0, _possibleConstructorReturn3.default)(this, (WrapComponent.__proto__ || (0, _getPrototypeOf2.default)(WrapComponent)).apply(this, arguments));
            }

            (0, _createClass3.default)(WrapComponent, [{
                key: 'handleChange',
                value: function handleChange() {
                    var _props = this.props,
                        editorState = _props.editorState,
                        onChange = _props.onChange;

                    onChange(_draftJs.RichUtils.toggleInlineStyle(editorState, style));
                }
            }, {
                key: 'render',
                value: function render() {
                    var currentStyle = this.props.editorState.getCurrentInlineStyle();
                    return _react2.default.createElement(WrapedComponent, (0, _extends3.default)({}, this.props, {
                        active: currentStyle.has(style),
                        onClick: this.handleChange.bind(this) }));
                }
            }]);
            return WrapComponent;
        }(_react.Component);

        WrapComponent.propTypes = {
            editorState: _propTypes2.default.object,
            onChange: _propTypes2.default.func
        };
        return WrapComponent;
    };
};

var BoldComponent = Wrap(_buttons.BoldButton)('BOLD');
var ItalicComponent = Wrap(_buttons.ItalicButton)('ITALIC');
var UnderlineComponent = Wrap(_buttons.UnderlineButton)('UNDERLINE');

exports.BoldComponent = BoldComponent;
exports.ItalicComponent = ItalicComponent;
exports.UnderlineComponent = UnderlineComponent;
