// carbon/lib/components/form/flavors/input-search/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Input = function (_Component) {
    (0, _inherits3.default)(Input, _Component);

    function Input() {
        (0, _classCallCheck3.default)(this, Input);
        return (0, _possibleConstructorReturn3.default)(this, (Input.__proto__ || (0, _getPrototypeOf2.default)(Input)).apply(this, arguments));
    }

    (0, _createClass3.default)(Input, [{
        key: 'handleSearchClick',
        value: function handleSearchClick(e) {
            this.props.onClickCallback(e);
        }
    }, {
        key: 'handleSearchChange',
        value: function handleSearchChange() {
            this.inputNode.focus();
            this.props.searchCallback(this.inputNode.value);
        }
    }, {
        key: 'clearClick',
        value: function clearClick() {
            this.inputNode.value = '';
        }
    }, {
        key: 'render',
        value: function render() {
            var _this2 = this;

            var _props = this.props,
                error = _props.error,
                search = _props.search,
                clear = _props.clear,
                searchCallback = _props.searchCallback,
                onClickCallback = _props.onClickCallback,
                others = (0, _objectWithoutProperties3.default)(_props, ['error', 'search', 'clear', 'searchCallback', 'onClickCallback']);

            var inputClassName = search === 'left' ? 'mcds-left__search' : 'mcds-input';
            var clearClassName = clear ? _react2.default.createElement(
                'span',
                { className: 'mcds-clear__right', onClick: this.clearClick.bind(this) },
                _react2.default.createElement('i', { className: 'mcds-icon__white mcds-icon__clear-solid-14' })
            ) : null;
            // 用来关闭浏览器默认的提示信息.TODO 其他的input是否以后要加上
            return _react2.default.createElement(
                'div',
                { className: (0, _classnames2.default)('mcds-input__container', this.props.className, { 'mcds-input__border': error }) },
                this.props.label ? _react2.default.createElement(
                    'label',
                    { className: 'mcds-label' },
                    this.props.label
                ) : null,
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-form__control' },
                    _react2.default.createElement('input', (0, _extends3.default)({}, others, {
                        className: inputClassName,
                        autoComplete: 'off',
                        ref: function ref(inputNode) {
                            _this2.inputNode = inputNode;
                        },
                        name: this.props.name,
                        type: 'text',
                        onChange: this.handleSearchChange.bind(this),
                        placeholder: this.props.placeholder })),
                    _react2.default.createElement(
                        'span',
                        { className: 'mcds-search__' + this.props.search, onClick: this.handleSearchClick.bind(this) },
                        _react2.default.createElement('i', { className: (0, _classnames2.default)(this.props.iconClass ? this.props.iconClass : 'mcds-icon__search-line-20') })
                    ),
                    clearClassName
                )
            );
        }
    }]);
    return Input;
}(_react.Component);

Input.propTypes = {
    label: _propTypes2.default.string,
    name: _propTypes2.default.string.isRequired,
    search: _propTypes2.default.oneOf(['left', 'right']),
    clear: _propTypes2.default.oneOf(['right']),
    className: _propTypes2.default.string,
    placeholder: _propTypes2.default.string,
    iconClass: _propTypes2.default.string,
    searchCallback: _propTypes2.default.func,
    onClickCallback: _propTypes2.default.func,
    error: _propTypes2.default.bool
};
Input.defaultProps = {
    name: 'name',
    search: null,
    clear: null,
    searchCallback: function searchCallback() {},
    onClickCallback: function onClickCallback() {}
};
exports.default = Input;
module.exports = exports['default'];
