// carbon/lib/components/editor/flavors/base/src/components/click-outside/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable */
/**
 * 监听这个组件外部的点击事件
 *
 * <ClickOutSide onClickOutside={() => console.log('click outside')}> <Component /> </ClickOutSide>
*/
var ClickOutSide = function (_Component) {
    (0, _inherits3.default)(ClickOutSide, _Component);

    function ClickOutSide() {
        (0, _classCallCheck3.default)(this, ClickOutSide);
        return (0, _possibleConstructorReturn3.default)(this, (ClickOutSide.__proto__ || (0, _getPrototypeOf2.default)(ClickOutSide)).apply(this, arguments));
    }

    (0, _createClass3.default)(ClickOutSide, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            document.addEventListener('click', this.handle.bind(this), true);
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            document.removeEventListener('click', this.handle.bind(this), true);
        }
    }, {
        key: 'handle',
        value: function handle(e) {
            var onClickOutside = this.props.onClickOutside;

            var el = this.refs.container;
            if (el && !el.contains(e.target)) {
                onClickOutside(e);
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var _props = this.props,
                children = _props.children,
                onClickOutside = _props.onClickOutside,
                props = (0, _objectWithoutProperties3.default)(_props, ['children', 'onClickOutside']);

            return _react2.default.createElement(
                'div',
                (0, _extends3.default)({}, props, { ref: 'container' }),
                children
            );
        }
    }]);
    return ClickOutSide;
}(_react.Component);

exports.default = ClickOutSide;


ClickOutSide.propTypes = {
    onClickOutside: _propTypes2.default.func.isRequired,
    children: _propTypes2.default.any
};
module.exports = exports['default'];
