// carbon/lib/components/file-selector/flavors/files/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.FileSelector = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FileSelector = function FileSelector(_ref) {
    var id = _ref.id,
        accept = _ref.accept,
        label = _ref.label,
        className = _ref.className,
        error = _ref.error,
        iconContent = _ref.iconContent,
        dropContent = _ref.dropContent,
        Icon = _ref.Icon,
        drop = _ref.drop,
        others = (0, _objectWithoutProperties3.default)(_ref, ['id', 'accept', 'label', 'className', 'error', 'iconContent', 'dropContent', 'Icon', 'drop']);
    return _react2.default.createElement(
        'div',
        { className: 'mcds-form__element' },
        label ? _react2.default.createElement(
            'span',
            { className: 'mcds-files__label' },
            label
        ) : null,
        _react2.default.createElement(
            'div',
            { className: 'mcds-form-element__control' },
            _react2.default.createElement(
                'div',
                { className: (0, _classnames2.default)('mcds-file__selector', className) },
                _react2.default.createElement(
                    'div',
                    (0, _extends3.default)({}, others, { className: (0, _classnames2.default)(' ', drop ? 'mcds-file-selector__dropzone' : null) }),
                    _react2.default.createElement(
                        'label',
                        { className: 'mcds-file-selector__body' },
                        _react2.default.createElement('input', { className: 'mcds-file-selector__input mcds-assistive__text', accept: accept, type: 'file', id: id }),
                        _react2.default.createElement(
                            'span',
                            { className: 'mcds-button mcds-button__currency' },
                            _react2.default.createElement('span', { className: (0, _classnames2.default)('mcds-vertical__sub mcds-text__size-20', Icon) }),
                            iconContent
                        ),
                        drop ? _react2.default.createElement(
                            'span',
                            { className: 'mcds-file-selector__text mcds-medium__show' },
                            dropContent
                        ) : null
                    )
                )
            )
        ),
        error ? _react2.default.createElement(
            'div',
            { className: 'mcds-form-element__help' },
            error
        ) : null
    );
};

exports.FileSelector = FileSelector;
FileSelector.propTypes = {
    id: _propTypes2.default.string,
    drop: _propTypes2.default.bool,
    accept: _propTypes2.default.string,
    label: _propTypes2.default.string,
    Icon: _propTypes2.default.string,
    className: _propTypes2.default.string,
    error: _propTypes2.default.string,
    iconContent: _propTypes2.default.string,
    dropContent: _propTypes2.default.string
};
