// carbon/lib/components/date-picker/flavor/_base/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.DatePickerBottomFunsColumn = exports.DatePickerBottomFuns = exports.DatePickerMonthCalendar = exports.DatePickerYearCalendar = exports.DatePickerDayCalendar = exports.DatePickerHeaderColumn = exports.DatePickerHeader = exports.DatePickerContainer = exports.langs = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 *  <DatePickerContainer>
 *      <DatePicker Header/>
 *      <Day Calendar /> || <Year Calendar /> || <Month Calender />
 *      <Extend Function Footer />
 *  </DatePickerContainer>
 */
var langs = exports.langs = {
    weeks: {
        mo: '一',
        tu: '二',
        we: '三',
        th: '四',
        fr: '五',
        sa: '六',
        su: '日'
    },
    year: '年',
    month: '月'
};

var DatePickerContainer = function DatePickerContainer(_ref) {
    var className = _ref.className,
        children = _ref.children,
        others = (0, _objectWithoutProperties3.default)(_ref, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-datepicker', className) }),
        children
    );
};

exports.DatePickerContainer = DatePickerContainer;
DatePickerContainer.propTypes = {
    children: _propTypes2.default.any.isRequired,

    className: _propTypes2.default.string,
    others: _propTypes2.default.any
};

var DatePickerHeader = function DatePickerHeader(_ref2) {
    var className = _ref2.className,
        children = _ref2.children,
        others = (0, _objectWithoutProperties3.default)(_ref2, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-datepicker__header', className) }),
        children
    );
};

exports.DatePickerHeader = DatePickerHeader;
DatePickerHeader.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string
};

var DatePickerHeaderColumn = function DatePickerHeaderColumn(_ref3) {
    var className = _ref3.className,
        children = _ref3.children,
        others = (0, _objectWithoutProperties3.default)(_ref3, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-datepicker__header-column', className) }),
        children
    );
};

exports.DatePickerHeaderColumn = DatePickerHeaderColumn;
DatePickerHeaderColumn.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string
};

var DatePickerDayCalendar = exports.DatePickerDayCalendar = function DatePickerDayCalendar(props) {
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, props, { className: (0, _classnames2.default)('mcds-datepicker__calendar mcds-datepicker__calendar-day') }),
        props.children
    );
};

DatePickerDayCalendar.propTypes = {
    children: _propTypes2.default.any.isRequired
};

var DatePickerYearCalendar = exports.DatePickerYearCalendar = function DatePickerYearCalendar(props) {
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, props, { className: (0, _classnames2.default)('mcds-datepicker__calendar mcds-datepicker__calendar-year', props.className) }),
        props.children
    );
};

DatePickerYearCalendar.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any.isRequired
};

// 日期控件 月份
var DatePickerMonthCalendar = function DatePickerMonthCalendar(_ref4) {
    var className = _ref4.className,
        children = _ref4.children,
        others = (0, _objectWithoutProperties3.default)(_ref4, ['className', 'children']);

    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-datepicker__calendar mcds-datepicker__calendar-month', className) }),
        children
    );
};

exports.DatePickerMonthCalendar = DatePickerMonthCalendar;
DatePickerMonthCalendar.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any.isRequired
};

// 日期控件 底部功能区容器
var DatePickerBottomFuns = exports.DatePickerBottomFuns = function DatePickerBottomFuns(props) {
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, props, { className: (0, _classnames2.default)('mcds-datepicker__foot mcds-datepicker__foot-funs', props.className) }),
        props.children
    );
};

DatePickerBottomFuns.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any.isRequired
};

var DatePickerBottomFunsColumn = exports.DatePickerBottomFunsColumn = function DatePickerBottomFunsColumn(props) {
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, props, { className: (0, _classnames2.default)('mcds-datepicker__foot-funs-column', props.className) }),
        props.children
    );
};

DatePickerBottomFunsColumn.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any.isRequired
};
