// carbon/lib/components/editor/flavors/base/src/plugins/font-size/panel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable */

var FONT_SIZE_MAP = [12, 13, 14, 16, 18, 20, 24];

var Panel = function (_Component) {
    (0, _inherits3.default)(Panel, _Component);

    function Panel() {
        (0, _classCallCheck3.default)(this, Panel);
        return (0, _possibleConstructorReturn3.default)(this, (Panel.__proto__ || (0, _getPrototypeOf2.default)(Panel)).apply(this, arguments));
    }

    (0, _createClass3.default)(Panel, [{
        key: 'render',
        value: function render() {
            var onClick = this.props.onClick;

            return _react2.default.createElement(
                'ul',
                { className: 'mcds-editor__font-size-root' },
                FONT_SIZE_MAP.map(function (c) {
                    return _react2.default.createElement(PanelItem, { key: 'fontSize' + c, val: c, onClick: onClick });
                })
            );
        }
    }]);
    return Panel;
}(_react.Component);

exports.default = Panel;


Panel.propTypes = {
    val: _propTypes2.default.object,
    onClick: _propTypes2.default.func
};
var PanelItem = function PanelItem(_ref) {
    var val = _ref.val,
        onClick = _ref.onClick;
    return _react2.default.createElement(
        'li',
        {
            onClick: onClick.bind(undefined, val),
            style: { fontSize: val + 'px' } },
        val,
        ' px'
    );
};
PanelItem.propTypes = {
    val: _propTypes2.default.number,
    onClick: _propTypes2.default.func
};
module.exports = exports['default'];
