// carbon/lib/components/dropdown/flavor/base.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DropDown = function DropDown(_ref) {
    var className = _ref.className,
        children = _ref.children,
        others = (0, _objectWithoutProperties3.default)(_ref, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-dropdown', className) }),
        children
    );
};

DropDown.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string
};

var DropDownList = function DropDownList(_ref2) {
    var className = _ref2.className,
        children = _ref2.children,
        others = (0, _objectWithoutProperties3.default)(_ref2, ['className', 'children']);
    return _react2.default.createElement(
        'ul',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-dropdown__list', className), role: 'menu' }),
        children
    );
};

DropDownList.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string
};

var DropDownItemHeader = function DropDownItemHeader(_ref3) {
    var className = _ref3.className,
        children = _ref3.children;
    return _react2.default.createElement(
        'li',
        { className: (0, _classnames2.default)('mcds-dropdown__item mcds-dropdown__header', className) },
        children
    );
};

DropDownItemHeader.propTypes = {
    children: _propTypes2.default.any,
    className: _propTypes2.default.string
};

var DropDownItem = function DropDownItem(props) {
    var iconLeft = props.iconLeft,
        iconRight = props.iconRight,
        isSelected = props.isSelected,
        tabIndex = props.tabIndex,
        className = props.className,
        others = (0, _objectWithoutProperties3.default)(props, ['iconLeft', 'iconRight', 'isSelected', 'tabIndex', 'className']);

    return _react2.default.createElement(
        'li',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-dropdown__item', className), role: 'presentation' }),
        _react2.default.createElement(
            'span',
            { className: (0, _classnames2.default)('mcds-dropdown__item-a', { selected: isSelected }), tabIndex: tabIndex || '-1' },
            _react2.default.createElement(
                'span',
                { className: 'mcds-truncate' },
                iconLeft ? iconLeft : null,
                props.children
            ),
            iconRight ? iconRight : null
        )
    );
};

DropDownItem.propTypes = {
    children: _propTypes2.default.any,
    disabled: _propTypes2.default.bool,
    iconLeft: _propTypes2.default.element,
    iconRight: _propTypes2.default.element,
    isSelected: _propTypes2.default.bool,
    tabIndex: _propTypes2.default.number,
    className: _propTypes2.default.string
};

var DropDownItemDivider = function DropDownItemDivider() {
    return _react2.default.createElement('li', { className: 'mcds-dropdown__divider', role: 'separator' });
};

exports.default = {
    DropDown: DropDown,
    DropDownItem: DropDownItem,
    DropDownList: DropDownList,
    DropDownItemHeader: DropDownItemHeader,
    DropDownItemDivider: DropDownItemDivider
};
module.exports = exports['default'];
