// carbon/lib/components/dropdown/flavor/dropdown-trigger/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _selfTrigger = require('./self-trigger');

var _selfTrigger2 = _interopRequireDefault(_selfTrigger);

var _bodyTrigger = require('./body-trigger');

var _bodyTrigger2 = _interopRequireDefault(_bodyTrigger);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Direction = ['left', 'right', 'bottom', 'top', 'top-left', 'top-right', 'right-top', 'right-bottom', 'bottom-left', 'bottom-right', 'left-top', 'left-bottom'];
// 定义位置

var DropDownTrigger = function (_React$Component) {
    (0, _inherits3.default)(DropDownTrigger, _React$Component);

    function DropDownTrigger() {
        (0, _classCallCheck3.default)(this, DropDownTrigger);

        var _this = (0, _possibleConstructorReturn3.default)(this, (DropDownTrigger.__proto__ || (0, _getPrototypeOf2.default)(DropDownTrigger)).call(this));

        _this.open = _this.open.bind(_this);
        _this.close = _this.close.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(DropDownTrigger, [{
        key: 'open',
        value: function open() {
            this.refs.trigger.open();
        }
    }, {
        key: 'close',
        value: function close() {
            this.refs.trigger.close();
        }
    }, {
        key: 'render',
        value: function render() {
            var _props = this.props,
                _props$target = _props.target,
                target = _props$target === undefined ? 'self' : _props$target,
                autoPlacement = _props.autoPlacement,
                others = (0, _objectWithoutProperties3.default)(_props, ['target', 'autoPlacement']);

            var ref = 'trigger';

            if (target === 'self') {
                return _react2.default.createElement(_selfTrigger2.default, (0, _extends3.default)({ ref: ref }, others));
            }

            if (target === 'body') {
                return _react2.default.createElement(_bodyTrigger2.default, (0, _extends3.default)({ ref: ref, autoPlacement: autoPlacement }, others));
            }

            return _react2.default.createElement('div', null);
        }
    }]);
    return DropDownTrigger;
}(_react2.default.Component);

exports.default = DropDownTrigger;


DropDownTrigger.propTypes = {
    target: _propTypes2.default.oneOf(['body', 'self']),
    placement: _propTypes2.default.oneOf(Direction),
    closeOnOutsideClick: _propTypes2.default.bool,
    autoPlacement: _propTypes2.default.bool,
    className: _propTypes2.default.string,
    autoCloseTag: _propTypes2.default.string,
    synchWidth: _propTypes2.default.bool, // 同步弹出层的宽度
    onClose: _propTypes2.default.func,
    onOpen: _propTypes2.default.func,
    offset: _propTypes2.default.number
};
DropDownTrigger.defaultProps = {
    onClose: function onClose() {
        // alert("onClose");
    },
    onOpen: function onOpen() {
        // alert("onOpen");
    },
    closeOnOutsideClick: true,
    autoPlacement: true,
    synchWidth: false,
    target: 'self',
    placement: 'bottom-right',
    offset: 5 // 弹出层距离target的距离
};
module.exports = exports['default'];
