// carbon/lib/components/color-picker/flavors/utils/Hue.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Hue = function (_React$Component) {
    (0, _inherits3.default)(Hue, _React$Component);

    function Hue() {
        (0, _classCallCheck3.default)(this, Hue);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Hue.__proto__ || (0, _getPrototypeOf2.default)(Hue)).call(this));

        _this.handleChange = _this.handleChange.bind(_this);
        _this.handleMouseDown = _this.handleMouseDown.bind(_this);
        _this.handleMouseUp = _this.handleMouseUp.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(Hue, [{
        key: 'handleChange',
        value: function handleChange(e, skip) {
            if (!skip) {
                e.preventDefault();
            }
            var container = this.refs.container;
            var containerWidth = container.clientWidth;
            var containerHeight = container.clientHeight;
            var left = (e.pageX || e.touches[0].pageX) - (container.getBoundingClientRect().left + window.pageXOffset);
            var top = (e.pageY || e.touches[0].pageY) - (container.getBoundingClientRect().top + window.pageYOffset);

            if (this.props.direction === 'vertical') {
                var h = void 0;
                if (top < 0) {
                    h = 359;
                } else if (top > containerHeight) {
                    h = 0;
                } else {
                    var percent = -(top * 100 / containerHeight) + 100;
                    h = 360 * percent / 100;
                }

                if (this.props.hsl.h !== h) {
                    this.props.onChange({
                        h: h,
                        s: this.props.hsl.s,
                        l: this.props.hsl.l,
                        a: this.props.hsl.a
                    });
                }
            } else {
                var _h = void 0;
                if (left < 0) {
                    _h = 0;
                } else if (left > containerWidth) {
                    _h = 359;
                } else {
                    var _percent = left * 100 / containerWidth;
                    _h = 360 * _percent / 100;
                }

                if (this.props.hsl.h !== _h) {
                    this.props.onChange({
                        h: _h,
                        s: this.props.hsl.s,
                        l: this.props.hsl.l,
                        a: this.props.hsl.a
                    });
                }
            }
        }
    }, {
        key: 'handleMouseDown',
        value: function handleMouseDown(e) {
            this.handleChange(e, true);
            window.addEventListener('mousemove', this.handleChange);
            window.addEventListener('mouseup', this.handleMouseUp);
        }
    }, {
        key: 'handleMouseUp',
        value: function handleMouseUp() {
            window.removeEventListener('mousemove', this.handleChange);
            window.removeEventListener('mouseup', this.handleMouseUp);
        }
    }, {
        key: 'render',
        value: function render() {
            var style = {};
            if (this.props.rgba === 'show') {
                style = {
                    hue: {
                        position: 'absolute',
                        top: '0px',
                        left: '0px',
                        right: '0px',
                        bottom: '0px',
                        background: 'linear-gradient(to right, #f00 0%, #ff0 17%, #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%)',
                        borderRadius: this.props.radius,
                        boxShadow: this.props.shadow
                    },
                    container: {
                        margin: '0 2px',
                        position: 'relative',
                        height: '100%'
                    },
                    pointer: {
                        zIndex: '2',
                        position: 'absolute',
                        left: this.props.hsl.h * 100 / 360 + '%'
                    },
                    slider: {
                        marginTop: '1px',
                        width: '4px',
                        borderRadius: '1px',
                        height: '8px',
                        boxShadow: '0 0 2px rgba(0, 0, 0, .6)',
                        background: '#fff',
                        transform: 'translateX(-2px)'
                    }
                };
            } else {
                style = {
                    hue: {
                        position: 'absolute',
                        top: '0px',
                        left: '0px',
                        right: '0px',
                        bottom: '0px',
                        background: 'linear-gradient(to right, #f00 0%, #ff0 17%, #0f0 33%, #0ff 50%, #00f 67%, #f0f 83%, #f00 100%)',
                        borderRadius: this.props.radius,
                        boxShadow: this.props.shadow
                    },
                    container: {
                        margin: '0 2px',
                        position: 'relative',
                        height: '16px'
                    },
                    pointer: {
                        zIndex: '2',
                        position: 'absolute',
                        left: this.props.hsl.h * 100 / 360 + '%'
                    },
                    slider: {
                        marginTop: '1px',
                        width: '4px',
                        borderRadius: '1px',
                        height: '14px',
                        boxShadow: '0 0 2px rgba(0, 0, 0, .6)',
                        background: '#fff',
                        transform: 'translateX(-2px)'
                    }
                };
            }
            var pointer = _react2.default.createElement('div', { style: style.slider });
            if (this.props.pointer) {
                pointer = _react2.default.createElement(this.props.pointer, this.props);
            }
            return _react2.default.createElement(
                'div',
                { style: style.hue },
                _react2.default.createElement(
                    'div',
                    { style: style.container, ref: 'container', onMouseDown: this.handleMouseDown, onTouchMove: this.handleChange },
                    _react2.default.createElement(
                        'div',
                        { style: style.pointer, ref: 'pointer' },
                        pointer
                    )
                )
            );
        }
    }]);
    return Hue;
}(_react2.default.Component);

Hue.propTypes = {
    onChange: _propTypes2.default.func,
    hsl: _propTypes2.default.object,
    hsv: _propTypes2.default.object,
    rgba: _propTypes2.default.string,
    radius: _propTypes2.default.string,
    shadow: _propTypes2.default.string,
    pointer: _propTypes2.default.object,
    direction: _propTypes2.default.string
};
exports.default = Hue;
module.exports = exports['default'];
