// carbon/lib/components/page-header/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./../index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * 存放示例
 */
var ObjectHome = function (_Component) {
    (0, _inherits3.default)(ObjectHome, _Component);

    function ObjectHome() {
        (0, _classCallCheck3.default)(this, ObjectHome);

        var _this = (0, _possibleConstructorReturn3.default)(this, (ObjectHome.__proto__ || (0, _getPrototypeOf2.default)(ObjectHome)).call(this));

        _this.state = {
            open_single: false
        };
        return _this;
    }

    (0, _createClass3.default)(ObjectHome, [{
        key: '_toggle',
        value: function _toggle() {
            this.setState({ open_single: !this.state.open_single });
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'div',
                { className: 'mcds-pageheader' },
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-grid mcds-pageheader__header' },
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-pageheader__header-left' },
                        _react2.default.createElement(
                            'div',
                            { className: 'mcds-media' },
                            _react2.default.createElement(
                                'div',
                                { className: 'mcds-m__r-30 mcds-p__t-6' },
                                _react2.default.createElement('div', { className: 'mcds-pageheader__header-left-icon' })
                            ),
                            _react2.default.createElement(
                                'div',
                                { className: 'mcds-media__body' },
                                _react2.default.createElement(
                                    'p',
                                    { className: 'mcds-pageheader__header-left-text' },
                                    '\u7EBF\u7D22'
                                ),
                                _react2.default.createElement(
                                    _index.DropDownTrigger,
                                    { className: '' },
                                    _react2.default.createElement(
                                        _index.Button,
                                        { className: 'mcds-text__line-28 mcds-pageheader__title' },
                                        '\u6211\u7684\u7EBF\u7D22',
                                        _react2.default.createElement('i', { className: 'mcds-icon__triangle-solid-14' })
                                    ),
                                    _react2.default.createElement(
                                        _index.DropDown,
                                        null,
                                        _react2.default.createElement(
                                            _index.DropDownList,
                                            null,
                                            _react2.default.createElement(
                                                _index.DropDownItemHeader,
                                                null,
                                                '\u7EBF\u7D22\u5217\u8868'
                                            ),
                                            _react2.default.createElement(
                                                _index.DropDownItem,
                                                null,
                                                '1'
                                            ),
                                            _react2.default.createElement(
                                                _index.DropDownItem,
                                                null,
                                                '2'
                                            ),
                                            _react2.default.createElement(
                                                _index.DropDownItem,
                                                null,
                                                '3'
                                            ),
                                            _react2.default.createElement(
                                                _index.DropDownItem,
                                                null,
                                                '4'
                                            )
                                        )
                                    )
                                )
                            )
                        )
                    ),
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-pageheader__header-right' },
                        _react2.default.createElement(
                            _index.ButtonGroup,
                            null,
                            _react2.default.createElement(
                                _index.Button,
                                { className: 'mcds-button__item' },
                                'New Lead'
                            ),
                            _react2.default.createElement(
                                _index.Button,
                                { className: 'mcds-button__item' },
                                'Import Leads'
                            ),
                            _react2.default.createElement(
                                _index.ButtonIcon,
                                { className: 'mcds-pageheader__header-right-icon', icon: 'mcds-icon__triangle-tiny' },
                                'Save'
                            )
                        )
                    )
                ),
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-grid mcds-pageheader__body' },
                    _react2.default.createElement(
                        'div',
                        null,
                        _react2.default.createElement(
                            _index.DropDownTrigger,
                            null,
                            _react2.default.createElement(_index.ButtonIcon, { hasDropdown: true, className: 'mcds-button-icon__more', icon: 'mcds-icon__settings' }),
                            _react2.default.createElement(
                                _index.DropDown,
                                { className: 'mcds-dropdown__position-right' },
                                _react2.default.createElement(
                                    _index.DropDownList,
                                    null,
                                    _react2.default.createElement(
                                        _index.DropDownItem,
                                        { isSelected: true },
                                        '\u88AB\u9009\u4E2D\u7684\u52A0\u5165isSelected\u5C5E\u6027'
                                    ),
                                    _react2.default.createElement(
                                        _index.DropDownItem,
                                        null,
                                        '2'
                                    ),
                                    _react2.default.createElement(
                                        _index.DropDownItem,
                                        null,
                                        '3'
                                    ),
                                    _react2.default.createElement(
                                        _index.DropDownItem,
                                        null,
                                        '4'
                                    )
                                )
                            )
                        ),
                        _react2.default.createElement(_index.ButtonIcon, { hasDropdown: true, className: 'mcds-button-icon__more', icon: 'mcds-icon__note-line' }),
                        _react2.default.createElement(_index.ButtonIcon, { className: 'mcds-button__item', icon: 'mcds-icon__delete-solid-14' }),
                        _react2.default.createElement(_index.ButtonIcon, { className: 'mcds-button__item', icon: 'mcds-icon__funnel-solid-14' }),
                        _react2.default.createElement(
                            _index.ButtonGroup,
                            null,
                            _react2.default.createElement(_index.ButtonIcon, { className: 'mcds-button__item', icon: 'mcds-icon__delete-solid-14' }),
                            _react2.default.createElement(_index.ButtonIcon, { className: 'mcds-button__item', icon: 'mcds-icon__funnel-solid-14' })
                        )
                    )
                ),
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-grid mcds-pageheader__footer' },
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-pageheader__footer-left' },
                        _react2.default.createElement(
                            'ul',
                            { className: 'mcds-list__horizontal mcds-text__weak mcds-text__size-12' },
                            _react2.default.createElement(
                                'li',
                                { className: 'mcds-list__item' },
                                '6\u4E2A\u9879\u76EE'
                            ),
                            _react2.default.createElement(
                                'li',
                                { className: 'mcds-list__item mcds-p__l-7 mcds-p__r-7' },
                                '\u2022'
                            ),
                            _react2.default.createElement(
                                'li',
                                { className: 'mcds-list__item' },
                                '\u6309\u540D\u79F0\u6392\u5E8F'
                            ),
                            _react2.default.createElement(
                                'li',
                                { className: 'mcds-list__item mcds-p__l-7 mcds-p__r-7' },
                                '\u2022'
                            ),
                            _react2.default.createElement(
                                'li',
                                { className: 'mcds-list__item' },
                                '\u6700\u540E\u66F4\u65B0'
                            ),
                            _react2.default.createElement(
                                'li',
                                { className: 'mcds-list__item mcds-p__l-5' },
                                '02/08/2017'
                            ),
                            _react2.default.createElement(
                                'li',
                                { className: 'mcds-list__item mcds-p__l-10' },
                                '15:40'
                            )
                        )
                    )
                )
            );
        }
    }]);
    return ObjectHome;
}(_react.Component);

var ObjectHomeCode = '\n    constructor(){\n        super();\n        this.state = {\n            open_single: false\n        };\n    }\n\n    _toggle() {\n        this.setState({open_single: !this.state.open_single});\n    }\n    render() {\n        return (\n            <div className="mcds-pageheader">\n                    <div className="mcds-grid mcds-pageheader__header">\n                        <div className="mcds-pageheader__header-left">\n                            <div className="mcds-media">\n                                <div className="mcds-media__figure mcds-p__t-6">\n                                    <div className="mcds-pageheader__header-left-icon" />\n                                </div>\n                                <div className="mcds-media__body">\n                                    <span className="mcds-pageheader__header-left-text">\u7EBF\u7D22</span>\n                                    <DropDownTrigger className=\'\'>\n                                        <Button className="mcds-text__line-28 mcds-pageheader__title">\n                                                \u6211\u7684\u7EBF\u7D22\n                                                <i className="mcds-icon__triangle-solid-14" />\n                                        </Button>\n                                        <DropDown>\n                                            <DropDownList>\n                                                <DropDownItemHeader >\u7EBF\u7D22\u5217\u8868</DropDownItemHeader>\n                                                <DropDownItem>\n                                                    1\n                                                </DropDownItem>\n                                                <DropDownItem>\n                                                    2\n                                                </DropDownItem>\n                                                <DropDownItem>\n                                                    3\n                                                </DropDownItem>\n                                                <DropDownItem>\n                                                    4\n                                                </DropDownItem>\n                                            </DropDownList>\n                                        </DropDown>\n                                    </DropDownTrigger>\n                                </div>\n                            </div>\n                        </div>\n                        <div className="mcds-pageheader__header-right">\n                            <ButtonGroup>\n                                <Button className="mcds-button__item">\n                                    New Lead\n                                </Button>\n                                <Button className="mcds-button__item">\n                                    Import Leads\n                                </Button>\n                                <ButtonIcon className="mcds-pageheader__header-right-icon" icon="mcds-icon__triangle-tiny">\n                                    Save\n                                </ButtonIcon>\n                            </ButtonGroup>\n                        </div>\n                    </div>\n                    <div className="mcds-grid mcds-pageheader__body">\n                        <div>\n                            <DropDownTrigger>\n                                <ButtonIcon hasDropdown={true} className="mcds-button-icon__more" icon="mcds-icon__settings" />\n                                <DropDown>\n                                    <DropDownList>\n                                        <DropDownItem isSelected={true}>\n                                            \u88AB\u9009\u4E2D\u7684\u52A0\u5165isSelected\u5C5E\u6027\n                                        </DropDownItem>\n                                        <DropDownItem>\n                                            2\n                                        </DropDownItem>\n                                        <DropDownItem>\n                                            3\n                                        </DropDownItem>\n                                        <DropDownItem>\n                                            4\n                                        </DropDownItem>\n                                    </DropDownList>\n                                </DropDown>\n                            </DropDownTrigger>\n                            \n                            <ButtonIcon hasDropdown={true} className="mcds-button-icon__more" icon="mcds-icon__note-line" />\n                            <ButtonIcon className="mcds-button-icon__more" icon="mcds-icon__edit-line" />\n                            <ButtonIcon className="mcds-button-icon__more" icon="mcds-icon__reopen-tiny" />\n                            <ButtonGroup>\n                                <ButtonIcon className="mcds-button-icon__more" icon="mcds-icon__binding-small" />\n                                <ButtonIcon className="mcds-button-icon__more" icon="mcds-icon__funnel-line" />\n                            </ButtonGroup>\n                        </div>\n                    </div>\n                    <div className="mcds-grid mcds-pageheader__footer">\n                        <div className="mcds-pageheader__footer-left">\n                            <ul className="mcds-list__horizontal mcds-text__weak mcds-text__size-12">\n                                <li className=\'mcds-list__item\'>6\u4E2A\u9879\u76EE</li>\n                                <li className=\'mcds-list__item mcds-p__l-7 mcds-p__r-7\'>\u2022</li>\n                                <li className=\'mcds-list__item\'>\u6309\u540D\u79F0\u6392\u5E8F</li>\n                                <li className=\'mcds-list__item mcds-p__l-7 mcds-p__r-7\'>\u2022</li>\n                                <li className=\'mcds-list__item\'>\u6700\u540E\u66F4\u65B0</li>\n                                <li className=\'mcds-list__item mcds-p__l-5\'>02/08/2017</li>\n                                <li className=\'mcds-list__item mcds-p__l-10\'>15:40</li>\n                            </ul>\n                        </div>\n                    </div>\n                </div>\n        );\n    }';

var RecordHome = function RecordHome() {
    return _react2.default.createElement(
        'div',
        { className: 'mcds-pageheader' },
        _react2.default.createElement(
            'div',
            { className: 'mcds-grid mcds-pageheader__header' },
            _react2.default.createElement(
                'div',
                { className: 'mcds-pageheader__header-left' },
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-media' },
                    _react2.default.createElement(
                        'div',
                        { style: { paddingTop: '6px' }, className: 'mcds-media__figure' },
                        _react2.default.createElement('div', { className: 'mcds-pageheader__header-left-icon' })
                    ),
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-media__body mcds-m__l-30' },
                        _react2.default.createElement(
                            'span',
                            { className: 'mcds-pageheader__header-left-text' },
                            'Record Type'
                        ),
                        _react2.default.createElement(
                            'div',
                            { className: 'mcds-pageheader__title' },
                            'Record Title'
                        )
                    )
                )
            ),
            _react2.default.createElement(
                'div',
                { className: 'mcds-pageheader__header-right' },
                _react2.default.createElement(
                    _index.StatefulButton,
                    null,
                    'Following'
                ),
                _react2.default.createElement(
                    _index.ButtonGroup,
                    null,
                    _react2.default.createElement(
                        _index.Button,
                        { className: 'mcds-button__item' },
                        'Edit'
                    ),
                    _react2.default.createElement(
                        _index.Button,
                        { className: 'mcds-button__item' },
                        'Delete'
                    ),
                    _react2.default.createElement(
                        _index.Button,
                        { className: 'mcds-button__item' },
                        'Clone'
                    ),
                    _react2.default.createElement(_index.ButtonIcon, { className: 'mcds-pageheader__header-right-icon', icon: 'mcds-icon__triangle-tiny' })
                )
            )
        ),
        _react2.default.createElement(
            'div',
            { className: 'mcds-m__t-30' },
            _react2.default.createElement(
                'table',
                { className: 'mcds-table mcds-table mcds-table-fixed__layout' },
                _react2.default.createElement(
                    'thead',
                    null,
                    _react2.default.createElement(
                        'tr',
                        null,
                        _react2.default.createElement(
                            _index.Th,
                            { className: 'mcds-is__sortable' },
                            'Field'
                        ),
                        _react2.default.createElement(
                            _index.Th,
                            { className: 'mcds-is__sortable', icon: 'mcds-icon__triangle-small' },
                            'Field(3)'
                        ),
                        _react2.default.createElement(
                            _index.Th,
                            { className: 'mcds-is__sortable' },
                            'Field'
                        ),
                        _react2.default.createElement(
                            _index.Th,
                            { className: 'mcds-is__sortable' },
                            'Field'
                        )
                    )
                ),
                _react2.default.createElement(
                    'tbody',
                    null,
                    _react2.default.createElement(
                        'tr',
                        null,
                        _react2.default.createElement(
                            'td',
                            { className: 'mcds-truncate' },
                            'Description tha demonstrates truncation with a lo\u2026'
                        ),
                        _react2.default.createElement(
                            'td',
                            { className: 'mcds-truncate' },
                            'Multiple Values'
                        ),
                        _react2.default.createElement(
                            'td',
                            { className: 'mcds-truncate' },
                            'Hyperlink'
                        ),
                        _react2.default.createElement(
                            'td',
                            { className: 'mcds-truncate' },
                            'Description (2-line truncat\u2026'
                        )
                    )
                )
            )
        )
    );
};

var RecordHomeCode = '\n            <div className="mcds-pageheader">\n            <div className="mcds-grid mcds-pageheader__header">\n                <div className="mcds-pageheader__header-left">\n                    <div className="mcds-media">\n                        <div style={{paddingTop: \'6px\'}} className="mcds-media__figure">\n                            <div className="mcds-pageheader__header-left-icon" />\n                        </div>\n                        <div className="mcds-media__body">\n                            <span className="mcds-pageheader__header-left-text">Record Type</span>\n                            <div className="mcds-pageheader__title">\n                                Record Title\n                            </div>\n                        </div>\n                    </div>\n                </div>\n                <div className="mcds-pageheader__header-right">\n                    <StatefulButton>\n                        Following\n                    </StatefulButton>\n                    <ButtonGroup>\n                        <Button className="mcds-button__item">\n                            Edit\n                        </Button>\n                        <Button className="mcds-button__item">\n                            Delete\n                        </Button>\n                        <Button className="mcds-button__item">\n                            Clone\n                        </Button>\n                        <ButtonIcon className="mcds-pageheader__header-right-icon" icon="mcds-icon__triangle-tiny" />\n                    </ButtonGroup>\n                </div>\n            </div>\n            <div className="mcds-m__t-30">\n                <table class="mcds-table mcds-table-fixed__layout">\n                    <thead>\n                        <tr>\n                            <Th className="mcds-is__sortable">\n                                Field\n                            </Th>\n                            <Th className="mcds-is__sortable" icon="mcds-icon__triangle-small">\n                                Field(3)\n                            </Th>\n                            <Th className="mcds-is__sortable">\n                                Field\n                            </Th>\n                            <Th className="mcds-is__sortable">\n                                Field\n                            </Th>\n                        </tr>\n                    </thead>\n                    <tbody>\n                        <tr>\n                            <td className="mcds-truncate">\n                                Description tha demonstrates truncation with a lo\u2026\n                            </td>\n                            <td className="mcds-truncate">\n                                Multiple Values\n                            </td>\n                            <td className="mcds-truncate">\n                                Hyperlink\n                            </td>\n                            <td className="mcds-truncate">\n                                Description (2-line truncat\u2026\n                            </td>\n                        </tr>\n                    </tbody>\n                </table>\n            </div>\n        </div>';

var RelatedList = function RelatedList() {
    return _react2.default.createElement(
        'div',
        { className: 'mcds-pageheader' },
        _react2.default.createElement(
            'div',
            { className: 'mcds-grid mcds-pageheader__header' },
            _react2.default.createElement(
                'div',
                { className: 'mcds-pageheader__header-left' },
                _react2.default.createElement(
                    _index.BreadCrumbs,
                    null,
                    _react2.default.createElement(
                        _index.Crumb,
                        null,
                        _react2.default.createElement(
                            'a',
                            { href: 'javascript:void(0);' },
                            'Accounts'
                        )
                    ),
                    _react2.default.createElement(
                        _index.Crumb,
                        null,
                        _react2.default.createElement(
                            'a',
                            { href: 'javascript:void(0);' },
                            'Company One'
                        )
                    )
                ),
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-pageheader__title' },
                    'Contacts (will truncate)'
                )
            ),
            _react2.default.createElement(
                'div',
                { className: 'mcds-pageheader__header-right' },
                _react2.default.createElement(
                    'div',
                    { className: 'pull-right mcds-m__b-10' },
                    _react2.default.createElement(
                        _index.ButtonGroup,
                        null,
                        _react2.default.createElement(
                            _index.Button,
                            { className: 'mcds-button__item' },
                            'Add Contact'
                        ),
                        _react2.default.createElement(_index.ButtonIcon, { className: 'mcds-pageheader__header-right-icon', icon: 'mcds-icon__triangle-tiny' })
                    )
                ),
                _react2.default.createElement(
                    'div',
                    null,
                    _react2.default.createElement(_index.ButtonIcon, { hasDropdown: true, className: 'mcds-button-icon__more', icon: 'mcds-icon__note-line' }),
                    _react2.default.createElement(
                        _index.ButtonGroup,
                        null,
                        _react2.default.createElement(_index.ButtonIcon, { className: 'mcds-button__item', icon: 'mcds-icon__delete-solid-14' }),
                        _react2.default.createElement(_index.ButtonIcon, { className: 'mcds-button__item', icon: 'mcds-icon__funnel-solid-14' }),
                        _react2.default.createElement(_index.ButtonIcon, { hasDropdown: true, className: 'mcds-button-icon__more', icon: 'mcds-icon__settings-line-20' })
                    )
                )
            )
        ),
        _react2.default.createElement(
            'div',
            { className: 'mcds-grid mcds-pageheader__footer' },
            _react2.default.createElement(
                'p',
                null,
                '10 items \u2022 Sorted by Name'
            )
        )
    );
};

var RelatedListCode = '\n            <div className="mcds-pageheader">\n            <div className="mcds-grid mcds-pageheader__header">\n                <div className="mcds-pageheader__header-left">\n                    <BreadCrumbs>\n                        <Crumb>\n                            <a href="javascript:void(0);">\n                                Accounts\n                            </a>\n                        </Crumb>\n                        <Crumb>\n                            <a href="javascript:void(0);">\n                                Company One\n                            </a>\n                        </Crumb>\n                    </BreadCrumbs>\n                    <div className="mcds-pageheader__title">\n                        Contacts (will truncate)\n                    </div>\n                </div>\n                <div className="mcds-pageheader__header-right" >\n                    <div style={{marginBottom: \'10px\'}} className="pull-right">\n                        <ButtonGroup>\n                            <Button className="mcds-button__item">\n                                Add Contact\n                            </Button>\n                            <ButtonIcon className="mcds-pageheader__header-right-icon" icon="mcds-icon__triangle-tiny" />\n                        </ButtonGroup>\n                    </div>\n                    <div>\n                        <ButtonIcon hasDropdown={true} className="mcds-button-icon__more" icon="mcds-icon__note-line" />\n                        <ButtonGroup>\n                            <ButtonIcon className="mcds-button-icon__more" icon="mcds-icon__binding-small" />\n                            <ButtonIcon className="mcds-button-icon__more" icon="mcds-icon__funnel-line" />\n                            <ButtonIcon hasDropdown={true} className="mcds-button-icon__more" icon="mcds-icon__settings" />\n                        </ButtonGroup>\n                    </div>\n                </div>\n            </div>\n            <div className="mcds-grid mcds-pageheader__footer">\n                <p>10 items \u2022 Sorted by Name</p>\n            </div>\n        </div>';

exports.default = [{
    id: 'ObjectHome',
    element: _react2.default.createElement(ObjectHome, null),
    intro: '有一些图标样式没有，用其他代替的，等设计出crm最新图标再换上;一些不确定的样式我用的行内样式来写的，没放到class中',
    code: ObjectHomeCode
}, {
    id: 'RecordHome',
    element: _react2.default.createElement(RecordHome, null),
    intro: 'Follow这个按钮样式hover会有些问题，table这里也是没有找到特别合适的样式，我修改了一下td的样式，改成单行文本溢出省略了',
    code: RecordHomeCode
}, {
    id: 'RelatedList',
    element: _react2.default.createElement(RelatedList, null),
    intro: 'RelatedListCode',
    code: RelatedListCode
}];
module.exports = exports['default'];
