// carbon/lib/components/lookup/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkdown = require('react-markdown');

var _reactMarkdown2 = _interopRequireDefault(_reactMarkdown);

var _index = require('./index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var objectList = [{
    icon: 'mcds-icon__telephone-solid-24',
    value: 'standard',
    content: '标准对象'
}, {
    icon: 'mcds-icon__telephone-solid-24',
    value: 'leads',
    content: '线索对象'
}];
var Data = [{
    id: 1,
    name: 'abc',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 2,
    name: 'abc123',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 3,
    name: 'abc123',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 4,
    name: 'abc123',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 5,
    name: 'abc',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 6,
    name: 'abc',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 71,
    name: 'abc',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 7,
    name: 'abc',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 9,
    name: 'abc',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 10,
    name: 'abc',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}];

var newDate = [{
    id: 1,
    name: 'abcfff',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}, {
    id: 2,
    name: 'abc',
    avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
}];

var DefaultLookup = function (_React$Component) {
    (0, _inherits3.default)(DefaultLookup, _React$Component);

    function DefaultLookup(props) {
        (0, _classCallCheck3.default)(this, DefaultLookup);

        var _this = (0, _possibleConstructorReturn3.default)(this, (DefaultLookup.__proto__ || (0, _getPrototypeOf2.default)(DefaultLookup)).call(this, props));

        _this.state = {
            data: Data,
            value: []
        };
        return _this;
    }

    (0, _createClass3.default)(DefaultLookup, [{
        key: 'toggleDefaultValue',
        value: function toggleDefaultValue() {
            var value = [{
                id: 2,
                name: Math.random().toString(),
                avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
            }];

            this.setState({
                value: value
            });
        }
    }, {
        key: 'resetDefaultValue',
        value: function resetDefaultValue() {
            this.setState({
                value: []
            });
        }
    }, {
        key: 'onKeyDown',
        value: function onKeyDown() {
            this.setState({ data: newDate });
        }
    }, {
        key: 'render',
        value: function render() {
            var value = this.state.value;
            var defaultValue = [{
                id: 2,
                name: '123',
                avatar: 'http://cdnimg103.lizhi.fm/audio_cover/2016/06/30/29596504232305031_320x320.jpg'
            }];
            return _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(
                    'button',
                    { onClick: this.resetDefaultValue.bind(this) },
                    'reset'
                ),
                _react2.default.createElement(
                    'button',
                    { onClick: this.toggleDefaultValue.bind(this) },
                    'toggle'
                ),
                _react2.default.createElement(_index.Lookup, {
                    error: false,
                    type: 'single',
                    onChange: function onChange(v) {
                        console.log('v', v);
                    },
                    defaultValue: defaultValue,
                    onKeyDown: this.onKeyDown.bind(this),
                    objectList: objectList,
                    data: this.state.data, value: value })
            );
        }
    }]);
    return DefaultLookup;
}(_react2.default.Component);

var defaultLookupCode = '\n\n';

var intro = '##### MediaObject 一个左右结构,自适应的结构\r\n\r\n *不提供任何组件* 只提供了一套样式.';
exports.default = [{
    id: 'default',
    element: _react2.default.createElement(DefaultLookup, null),
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: intro }),
    code: defaultLookupCode
}];
module.exports = exports['default'];
