// carbon/lib/components/file-selector/flavors/integrated/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.IntegratedFileSelector = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var IntegratedFileSelector = function IntegratedFileSelector(_ref) {
    var id = _ref.id,
        accept = _ref.accept,
        className = _ref.className,
        drag = _ref.drag,
        draggover = _ref.draggover,
        draggoverError = _ref.draggoverError,
        children = _ref.children,
        errorMsg = _ref.errorMsg,
        dropContent = _ref.dropContent,
        others = (0, _objectWithoutProperties3.default)(_ref, ['id', 'accept', 'className', 'drag', 'draggover', 'draggoverError', 'children', 'errorMsg', 'dropContent']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-file-selector mcds-file-selector-integrated', className) }),
        _react2.default.createElement(
            'div',
            { className: (0, _classnames2.default)('mcds-file-selector__dropzone mcds-file-selector__dropzone-integrated', drag ? 'mcds-has-drag' : null, draggover ? 'mcds-has-drag-over' : null), 'aria-hidden': 'true' },
            _react2.default.createElement(
                'label',
                { className: 'mcds-file-selector__body mcds-file-selector__body-integrated' },
                _react2.default.createElement('input', { className: 'mcds-file-selector__input mcds-assistive-text', accept: accept || 'image/png', type: 'file', id: id, disabled: draggoverError }),
                draggoverError ? _react2.default.createElement('span', { className: 'mcds-text__size-40 mcds-icon__prohibit-line-20' }) : _react2.default.createElement('span', { className: 'mcds-text__size-40 mcds-icon__upload-line-20' }),
                _react2.default.createElement(
                    'span',
                    { className: 'mcds-file-selector__text mcds-file-selector__text-integrated mcds-text-heading-medium mcds-text-align-center' },
                    draggoverError ? errorMsg : dropContent
                )
            )
        ),
        children
    );
};

exports.IntegratedFileSelector = IntegratedFileSelector;
IntegratedFileSelector.propTypes = {
    id: _propTypes2.default.string,
    accept: _propTypes2.default.string,
    errorMsg: _propTypes2.default.string,
    className: _propTypes2.default.string,
    drag: _propTypes2.default.bool,
    draggover: _propTypes2.default.bool,
    draggoverError: _propTypes2.default.bool,
    children: _propTypes2.default.any,
    dropContent: _propTypes2.default.string
};
