// carbon/lib/components/color-picker/flavors/base/sketchFields.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _color = require('./../utils/color');

var _color2 = _interopRequireDefault(_color);

var _utils = require('./../utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SketchFields = function (_React$Component) {
    (0, _inherits3.default)(SketchFields, _React$Component);

    function SketchFields() {
        (0, _classCallCheck3.default)(this, SketchFields);

        var _this = (0, _possibleConstructorReturn3.default)(this, (SketchFields.__proto__ || (0, _getPrototypeOf2.default)(SketchFields)).call(this));

        _this.handleChange = _this.handleChange.bind(_this);
        return _this;
    }
    // 直接输入十六进位hex值


    (0, _createClass3.default)(SketchFields, [{
        key: 'handleChange',
        value: function handleChange(data) {
            if (data.hex) {
                if (_color2.default.isValidHex(data.hex)) {
                    this.props.onChange(data.hex);
                }
            } else if (data.r || data.g || data.b) {
                this.props.onChange({
                    r: data.r || this.props.rgb.r,
                    g: data.g || this.props.rgb.g,
                    b: data.b || this.props.rgb.b,
                    a: this.props.rgb.a
                });
            } else if (data.a) {
                if (data.a < 0) {
                    data.a = 0;
                } else if (data.a > 100) {
                    data.a = 100;
                }
                data.a = data.a / 100;
                this.props.onChange({
                    h: this.props.hsl.h,
                    s: this.props.hsl.s,
                    l: this.props.hsl.l,
                    a: data.a
                });
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var style = {
                fields: {
                    display: 'flex',
                    paddingTop: '4px'
                },
                single: {
                    flex: '1',
                    paddingLeft: '6px'
                },
                double: {
                    flex: '2'
                },
                hexHolder: {
                    width: '100%'
                },
                only: {
                    flex: '1'
                }
            };
            var elem = _react2.default.createElement(
                'div',
                { style: style.fields, className: 'flexbox-fix' },
                _react2.default.createElement(
                    'div',
                    { style: style.double },
                    _react2.default.createElement(_utils.EditableInput, { label: 'hex', value: this.props.hex.replace('#', ''), onChange: this.handleChange })
                )
            );
            return elem;
        }
    }]);
    return SketchFields;
}(_react2.default.Component);

SketchFields.propTypes = {
    onChange: _propTypes2.default.func,
    hsl: _propTypes2.default.object,
    rgb: _propTypes2.default.object,
    rgba: _propTypes2.default.string,
    hex: _propTypes2.default.string
};
exports.default = SketchFields;
module.exports = exports['default'];
