// carbon/lib/components/color-picker/flavors/base/sketch.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _utils = require('./../utils');

var _sketchFields = require('./sketchFields');

var _sketchFields2 = _interopRequireDefault(_sketchFields);

var _sketchPresetColors = require('./sketchPresetColors');

var _sketchPresetColors2 = _interopRequireDefault(_sketchPresetColors);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Sketch = function (_React$Component) {
    (0, _inherits3.default)(Sketch, _React$Component);

    function Sketch() {
        (0, _classCallCheck3.default)(this, Sketch);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Sketch.__proto__ || (0, _getPrototypeOf2.default)(Sketch)).call(this));

        _this.handleChange = _this.handleChange.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(Sketch, [{
        key: 'handleChange',
        value: function handleChange(data) {
            this.props.onChange(data);
        }
    }, {
        key: 'render',
        value: function render() {
            var style = {
                picker: {
                    width: '200px',
                    padding: '10px 10px 0',
                    boxSizing: 'initial',
                    background: '#fff',
                    borderRadius: '4px',
                    boxShadow: '0 0 0 1px rgba(0,0,0,.15), 0 8px 16px rgba(0,0,0,.15)'
                },
                saturation: {
                    width: '100%',
                    paddingBottom: '75%',
                    position: 'relative',
                    overflow: 'hidden'
                },
                Saturation: {
                    radius: '3px',
                    shadow: 'inset 0 0 0 1px rgba(0,0,0,.15), inset 0 0 4px rgba(0,0,0,.25)'
                },
                controls: {
                    display: 'flex'
                },
                sliders: {
                    padding: '4px 0',
                    flex: '1'
                },
                color: {
                    width: '16px',
                    height: '16px',
                    position: 'relative',
                    marginTop: '4px',
                    marginLeft: '4px',
                    borderRadius: '3px'
                },
                activeColor: {
                    position: 'absolute',
                    top: '0px',
                    left: '0px',
                    right: '0px',
                    bottom: '0px',
                    borderRadius: '2px',
                    background: 'rgba(' + this.props.rgb.r + ', ' + this.props.rgb.g + ', ' + this.props.rgb.b + ', ' + this.props.rgb.a + ')',
                    boxShadow: 'inset 0 0 0 1px rgba(0,0,0,.15), inset 0 0 4px rgba(0,0,0,.25)',
                    zIndex: '2'
                },
                hue: {
                    position: 'relative',
                    height: '16px',
                    overflow: 'hidden'
                },
                Hue: {
                    radius: '2px',
                    shadow: 'inset 0 0 0 1px rgba(0,0,0,.15), inset 0 0 4px rgba(0,0,0,.25)'
                },
                alpha: {
                    display: 'none'
                },
                Alpha: {
                    radius: '2px',
                    shadow: 'inset 0 0 0 1px rgba(0,0,0,.15), inset 0 0 4px rgba(0,0,0,.25)'
                }
            };
            return _react2.default.createElement(
                'div',
                { style: style.picker },
                _react2.default.createElement(
                    'div',
                    { style: style.saturation },
                    _react2.default.createElement(_utils.Saturation, (0, _extends3.default)({ style: style.Saturation }, this.props, { onChange: this.handleChange }))
                ),
                _react2.default.createElement(
                    'div',
                    { style: style.controls },
                    _react2.default.createElement(
                        'div',
                        { style: style.sliders },
                        _react2.default.createElement(
                            'div',
                            { style: style.hue },
                            _react2.default.createElement(_utils.Hue, (0, _extends3.default)({ style: style.Hue }, this.props, { onChange: this.handleChange }))
                        ),
                        _react2.default.createElement(
                            'div',
                            { style: style.alpha },
                            _react2.default.createElement(_utils.Alpha, (0, _extends3.default)({ style: style.Alpha }, this.props, { onChange: this.handleChange }))
                        )
                    ),
                    _react2.default.createElement(
                        'div',
                        { style: style.color },
                        _react2.default.createElement('div', { style: style.activeColor }),
                        _react2.default.createElement(_utils.Checkboard, null)
                    )
                ),
                _react2.default.createElement(
                    'div',
                    { style: style.fields },
                    _react2.default.createElement(_sketchFields2.default, (0, _extends3.default)({}, this.props, { onChange: this.handleChange }))
                ),
                _react2.default.createElement(
                    'div',
                    { style: style.presets },
                    _react2.default.createElement(_sketchPresetColors2.default, { colors: this.props.presetColors, onClick: this.handleChange })
                )
            );
        }
    }]);
    return Sketch;
}(_react2.default.Component);

Sketch.propTypes = {
    a: _propTypes2.default.number,
    onChange: _propTypes2.default.func,
    hsl: _propTypes2.default.object,
    rgb: _propTypes2.default.object,
    width: _propTypes2.default.number,
    presetColors: _propTypes2.default.array
};
Sketch.defaultProps = {
    presetColors: ['#D0021B', '#F5A623', '#F8E71C', '#8B572A', '#7ED321', '#417505', '#BD10E0', '#9013FE', '#4A90E2', '#50E3C2', '#B8E986', '#000000', '#4A4A4A', '#9B9B9B', '#FFFFFF'],
    width: 200
};
exports.default = Sketch;
module.exports = exports['default'];
