// carbon/lib/components/process/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactMarkdown = require('react-markdown');

var _reactMarkdown2 = _interopRequireDefault(_reactMarkdown);

var _index = require('./index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stepArr = [{ text: 'Unqualified' }, { text: 'New' }, { text: 'Working' }, { text: 'Nurturing' }, { text: 'Closed', className: 'mcds-process__closed-lost' }];

var Example = function (_Component) {
    (0, _inherits3.default)(Example, _Component);

    function Example(props) {
        (0, _classCallCheck3.default)(this, Example);
        return (0, _possibleConstructorReturn3.default)(this, (Example.__proto__ || (0, _getPrototypeOf2.default)(Example)).call(this, props));
    }

    (0, _createClass3.default)(Example, [{
        key: 'handleBefore',
        value: function handleBefore(i, j) {
            console.log('onBefore', i, j);
            if (i === 3) {
                console.log('this is 3', i);
                if (i === 1) {
                    return true;
                }
                return false;
            }
            return true;
        }
    }, {
        key: 'handleAfter',
        value: function handleAfter(i, j) {
            console.log('onAfter', i, j);
        }
    }, {
        key: 'render',
        value: function render() {
            var section = [_react2.default.createElement(
                'section',
                { className: 'mcds-process__tab-grid', key: '123' },
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-layout__column mcds-divider__top mcds-divider__bottom mcds-process__tab-grid-item mcds-text__weak' },
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-layout__item' },
                        '\u5173\u952E\u5B57\u6BB5'
                    ),
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-layout__right' },
                        _react2.default.createElement(
                            'span',
                            { className: 'mcds-text__link mcds-cursor__pointer mcds-text__size-13' },
                            '\u7F16\u8F91'
                        )
                    )
                ),
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-layout__column mcds-layout__middle mcds-divider__bottom mcds-process__tab-grid-item mcds-text__weak' },
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-layout__item ' },
                        '\u90AE\u7BB1'
                    ),
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-layout__right mcds-text mcds-text__size-13' },
                        _react2.default.createElement(
                            'span',
                            { className: 'mcds-readonly__span' },
                            'ppp@ppp1.com'
                        )
                    )
                ),
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-layout__column mcds-layout__middle mcds-divider__bottom  mcds-process__tab-grid-item mcds-text__weak' },
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-layout__item' },
                        '\u5458\u5DE5\u6570'
                    ),
                    _react2.default.createElement(
                        'div',
                        { className: 'mcds-layout__right' },
                        _react2.default.createElement('i', { className: 'mcds-icon__edit-line-20 mcds-text__size-20 mcds-cursor__pointer' })
                    )
                )
            ), _react2.default.createElement(
                'section',
                { className: 'mcds-process__tab-grid  mcds-text__weak', key: '456' },
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-divider__top mcds-divider__bottom mcds-process__tab-grid-item' },
                    '\u6307\u5BFC'
                ),
                _react2.default.createElement(
                    'div',
                    { className: 'mcds-p__l-20 mcds-p__t-20 mcds-p__b-20 mcds-text__size-13 mcds-divider__bottom mcds-process__tab-grid-richtext' },
                    'Regularly cross-sell related products ',
                    _react2.default.createElement(
                        'a',
                        { href: 'javascript:void(0);' },
                        'using cross-sell tactics and principles.'
                    ),
                    ' Prepare demo deck using the ',
                    _react2.default.createElement(
                        'a',
                        { href: 'javascript:void(0);' },
                        'latest template'
                    ),
                    ' and review with Marketing and Sales teams. Review demo copy with Legal and Doc team.'
                )
            )];
            return _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(
                    _index.Process,
                    {
                        stepArr: stepArr,
                        currentStep: 2,
                        activeStep: 1,
                        onBefore: this.handleBefore,
                        onAfter: this.handleAfter,
                        isOpen: true,
                        haveButton: true },
                    _react2.default.createElement(
                        _index.ProcessTabContent,
                        null,
                        section
                    ),
                    _react2.default.createElement(
                        _index.ProcessTabContent,
                        null,
                        section
                    ),
                    _react2.default.createElement(
                        _index.ProcessTabContent,
                        null,
                        section
                    ),
                    _react2.default.createElement(
                        _index.ProcessTabContent,
                        null,
                        section
                    ),
                    _react2.default.createElement(
                        _index.ProcessTabContent,
                        null,
                        section
                    )
                )
            );
        }
    }]);
    return Example;
}(_react.Component);

var intro = '\r\n#### Process 进度条 支持传入\r\n\r\n-  currentStep 进度条当前状态（unmber类型，初始状态为0，Won状态为进度条项目数，Lose状态为-1）\r\n-  buttonText 右侧按钮的文本内容\r\n-  stepArr 可传入每个选项框的属性 text和className\r\n-  onBefore 进入其他进度前调用的函数，需返回布尔值判断是否进入，参数依次为activeStep和currentStep，即当前要进入的步骤（当前活动的步骤）和此时进度条所在的步骤\r\n- 这里的例子为进度位于1（第二个格子），活动模块位于3（第四个格子），若上方input为空，则进度不能跳到3\r\n- 增加isOpentab页内容是否开启 （布尔值，默认为开启）\r\n- 增加haveButton是否有右侧按钮 （布尔值，默认为有）';
exports.default = [{
    id: 'Process',
    element: _react2.default.createElement(Example, null),
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: intro }),
    code: '123'
}];
module.exports = exports['default'];
