// carbon/lib/components/tab/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _index = require('./index');

var _reactMarkdown = require('react-markdown');

var _reactMarkdown2 = _interopRequireDefault(_reactMarkdown);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var element = _react2.default.createElement(
    'div',
    { className: 'mcds-container' },
    _react2.default.createElement(
        _index.Tab,
        { className: 'mcds-tab__default' },
        _react2.default.createElement(
            _index.TabItem,
            null,
            _react2.default.createElement('span', { className: 'mcds-tab-left__icon mcds-icon__open-folder' }),
            'Item One',
            _react2.default.createElement('span', { className: 'mcds-tab-right__icon mcds-icon__close-medium' })
        ),
        _react2.default.createElement(
            _index.TabItem,
            { className: 'mcds-tab__active' },
            _react2.default.createElement('span', { className: 'mcds-tab-left__icon mcds-icon__open-folder' }),
            'Item Tow',
            _react2.default.createElement('span', { className: 'mcds-tab-right-small__icon mcds-icon-default__color mcds-icon__triangle-small' }),
            _react2.default.createElement('span', { className: 'mcds-tab-right__icon mcds-icon-default__color mcds-icon__close-medium' })
        ),
        _react2.default.createElement(
            _index.TabItem,
            null,
            'Item Three'
        ),
        _react2.default.createElement(
            _index.TabItem,
            null,
            'More ',
            _react2.default.createElement('span', { className: 'mcds-tab-right-small__icon mcds-icon__triangle-small' })
        ),
        _react2.default.createElement(
            _index.TabContent,
            { className: 'mcds-tab__active' },
            'Item Three Content'
        ),
        _react2.default.createElement(
            _index.TabContent,
            null,
            'Item Three Content'
        ),
        _react2.default.createElement(
            _index.TabContent,
            null,
            'Item Three Content'
        )
    )
);

var Scoped = _react2.default.createElement(
    'div',
    { className: 'mcds-container' },
    _react2.default.createElement(
        _index.Tab,
        { className: 'mcds-tab__scoped' },
        _react2.default.createElement(
            _index.TabItem,
            null,
            'Item Tow'
        ),
        _react2.default.createElement(
            _index.TabItem,
            { className: 'mcds-scoped__active' },
            'Item Three'
        ),
        _react2.default.createElement(
            _index.TabItem,
            null,
            'Item Three'
        ),
        _react2.default.createElement(
            _index.TabItem,
            null,
            'Item Three'
        ),
        _react2.default.createElement(
            _index.TabContent,
            { className: 'mcds-content__scoped mcds-scoped__active' },
            'Item Three Content'
        )
    )
);

var ElementScoped = _react2.default.createElement(
    'div',
    { className: 'mcds-container' },
    _react2.default.createElement(
        _index.Tab,
        { className: 'mcds-tab__default' },
        _react2.default.createElement(
            _index.TabItem,
            { className: 'mcds-tab__active' },
            'Item One'
        ),
        _react2.default.createElement(
            _index.TabItem,
            null,
            'Item Tow'
        ),
        _react2.default.createElement(
            _index.TabItem,
            null,
            'Item Three'
        ),
        _react2.default.createElement(
            _index.TabContent,
            { className: 'mcds-tab-scoped__padding mcds-tab__active' },
            _react2.default.createElement(
                _index.Tab,
                { className: 'mcds-tab__scoped' },
                _react2.default.createElement(
                    _index.TabItem,
                    null,
                    'Item Tow'
                ),
                _react2.default.createElement(
                    _index.TabItem,
                    { className: 'mcds-scoped__active' },
                    'Item Three'
                ),
                _react2.default.createElement(
                    _index.TabItem,
                    null,
                    'Item Three'
                ),
                _react2.default.createElement(
                    _index.TabItem,
                    null,
                    'Item Three'
                ),
                _react2.default.createElement(
                    _index.TabContent,
                    { className: 'mcds-content__scoped mcds-scoped__active' },
                    'Item Three Content'
                )
            )
        )
    )
);

var TabChange = function (_React$Component) {
    (0, _inherits3.default)(TabChange, _React$Component);

    function TabChange() {
        (0, _classCallCheck3.default)(this, TabChange);

        var _this = (0, _possibleConstructorReturn3.default)(this, (TabChange.__proto__ || (0, _getPrototypeOf2.default)(TabChange)).call(this));

        _this.state = {
            index: 2
        };
        return _this;
    }

    (0, _createClass3.default)(TabChange, [{
        key: 'changeItem',
        value: function changeItem(i) {
            this.setState({ index: i });
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                _index.Tab,
                { className: 'mcds-tab__default' },
                _react2.default.createElement(
                    _index.TabItem,
                    { className: (0, _classnames2.default)(this.state.index === 1 ? 'mcds-tab__active' : null), onClick: this.changeItem.bind(this, 1) },
                    _react2.default.createElement('span', { className: 'mcds-tab-left__icon mcds-icon__open-folder' }),
                    'Item One',
                    _react2.default.createElement('span', { className: 'mcds-tab-right__icon mcds-icon__close-medium' })
                ),
                _react2.default.createElement(
                    _index.TabItem,
                    { className: (0, _classnames2.default)(this.state.index === 2 ? 'mcds-tab__active' : null), onClick: this.changeItem.bind(this, 2) },
                    _react2.default.createElement('span', { className: 'mcds-tab-left__icon mcds-icon__open-folder' }),
                    'Item Tow',
                    _react2.default.createElement('span', { className: 'mcds-tab-right-small__icon mcds-icon-default__color mcds-icon__triangle-small' }),
                    _react2.default.createElement('span', { className: 'mcds-tab-right__icon mcds-icon-default__color mcds-icon__close-medium' })
                ),
                _react2.default.createElement(
                    _index.TabItem,
                    { className: (0, _classnames2.default)(this.state.index === 3 ? 'mcds-tab__active' : null), onClick: this.changeItem.bind(this, 3) },
                    'Item Three'
                ),
                _react2.default.createElement(
                    _index.TabItem,
                    { className: (0, _classnames2.default)(this.state.index === 4 ? 'mcds-tab__active' : null), onClick: this.changeItem.bind(this, 4) },
                    'More ',
                    _react2.default.createElement('span', { className: 'mcds-tab-right-small__icon mcds-icon__triangle-small' })
                ),
                _react2.default.createElement(
                    _index.TabContent,
                    { className: (0, _classnames2.default)(this.state.index === 1 ? 'mcds-tab__active' : null) },
                    'Item One Content'
                ),
                _react2.default.createElement(
                    _index.TabContent,
                    { className: (0, _classnames2.default)(this.state.index === 2 ? 'mcds-tab__active' : null) },
                    'Item Tow Content'
                ),
                _react2.default.createElement(
                    _index.TabContent,
                    { className: (0, _classnames2.default)(this.state.index === 3 ? 'mcds-tab__active' : null) },
                    'Item Three Content'
                ),
                _react2.default.createElement(
                    _index.TabContent,
                    { className: (0, _classnames2.default)(this.state.index === 4 ? 'mcds-tab__active' : null) },
                    'Item More Content'
                )
            );
        }
    }]);
    return TabChange;
}(_react2.default.Component);

var TabChangeDemo = '\nclass TabChange extends React.Component {\n    constructor() {\n        super();\n        this.state = {\n            index: 2\n        }\n    }\n    changeItem(i){\n        this.setState({index: i});\n    }\n    render() {\n        return (\n            <Tab className="mcds-tab__default">\n                <TabItem className={classnames(this.state.index === 1 ? \'mcds-tab__active\' : null )} onClick={this.changeItem.bind(this,1)}>\n                    <span className="mcds-tab-left__icon mcds-icon__open-folder" />\n                    Item One\n                    <span className="mcds-tab-right__icon mcds-icon__close-medium" />\n                </TabItem>\n                <TabItem className={classnames(this.state.index === 2 ? \'mcds-tab__active\' : null )} onClick={this.changeItem.bind(this,2)}>\n                    <span className="mcds-tab-left__icon mcds-icon__open-folder" />\n                    Item Tow\n                    <span className="mcds-tab-right-small__icon mcds-icon-default__color mcds-icon__triangle-small" />\n                    <span className="mcds-tab-right__icon mcds-icon-default__color mcds-icon__close-medium" />\n                </TabItem>\n                <TabItem className={classnames(this.state.index === 3 ? \'mcds-tab__active\' : null )} onClick={this.changeItem.bind(this,3)}>\n                    Item Three\n                </TabItem>\n                <TabItem className={classnames(this.state.index === 4 ? \'mcds-tab__active\' : null )} onClick={this.changeItem.bind(this,4)}>\n                    More <span className="mcds-tab-right-small__icon mcds-icon__triangle-small" />\n                </TabItem>\n\n                <TabContent className={classnames(this.state.index === 1 ? \'mcds-tab__active\' : null )}>\n                    Item One Content\n                </TabContent>\n                <TabContent className={classnames(this.state.index === 2 ? \'mcds-tab__active\' : null )}>\n                    Item Tow Content\n                </TabContent>\n                <TabContent className={classnames(this.state.index === 3 ? \'mcds-tab__active\' : null )}>\n                    Item Three Content\n                </TabContent>\n                <TabContent className={classnames(this.state.index === 4 ? \'mcds-tab__active\' : null )}>\n                    Item More Content\n                </TabContent>\n            </Tab>\n        );\n    }\n}';

exports.default = [{
    id: 'Tabs',
    element: element,
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: '#### Tab 有两种样式\r\n* 一种为 mcds-tab__default 另一种mcds-tab__scoped\r\n* mcds-tab__default 选中样式为mcds-tab__active\r\n* mcds-tab__scoped 选中样式为mcds-scoped__active\r\n* mcds-tab__scoped TabContent需要加类名 mcds-content__scoped\r\n* TabItem 与 TabContent的关系可以使一对多，也可以是一对一\r\n\r\n' })
}, {
    id: 'TabChangeDemo',
    element: _react2.default.createElement(TabChange, null),
    code: TabChangeDemo,
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: '#### 点击切换Demo' })
}, {
    id: 'Scoped',
    element: Scoped
}, {
    id: 'elementScoped',
    element: ElementScoped
}];
module.exports = exports['default'];
