// carbon/lib/components/date-picker/flavor/base/month-calendar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _base = require('../_base');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/*
    <DayCalendar
        init = {
            year:2016,
            month:1,
            day:1,
        }|| timestamp;
    />
 */
var Cell = function Cell(props) {
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, props, { className: (0, _classnames2.default)('mcds-datepicker__calendar-month-onemonth', props.className) }),
        props.children
    );
};

Cell.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string
};

var monthCalendar = function (_React$Component) {
    (0, _inherits3.default)(monthCalendar, _React$Component);

    function monthCalendar(props) {
        (0, _classCallCheck3.default)(this, monthCalendar);

        var _this = (0, _possibleConstructorReturn3.default)(this, (monthCalendar.__proto__ || (0, _getPrototypeOf2.default)(monthCalendar)).call(this, props));

        var init = {};

        if (typeof props.init === 'number') {
            var _init = (0, _moment2.default)(props.init);
            init = {
                year: _init.year(),
                month: _init.month(),
                day: _init.day()
            };
        } else {
            init = props.init;
        }

        _this.state = {
            init: init
        };
        _this.checkIsNow = _this.checkIsNow.bind(_this);
        _this.checkSelected = _this.checkSelected.bind(_this);

        _this.cellClick = _this.cellClick.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(monthCalendar, [{
        key: 'checkIsNow',
        value: function checkIsNow(month) {
            var now = (0, _moment2.default)();
            var year = this.state.init.year;

            if (now.month() === month && now.year() === year) {
                return true;
            }
            return false;
        }
    }, {
        key: 'checkSelected',
        value: function checkSelected(month) {
            if (month === this.state.init.month + 1) {
                return true;
            }
            return false;
        }
    }, {
        key: 'cellClick',
        value: function cellClick(month) {
            var year = this.state.init.year;

            this.props.cellChanged({ year: year, month: month });
        }
    }, {
        key: 'renderMonths',
        value: function renderMonths() {
            var _this2 = this;

            var month = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
            var arrs = [];
            var component = [];
            // 每七天分为一组week
            month.forEach(function (v, index, _month) {
                if (index % 4 === 0) {
                    var one_row = [];
                    for (var i = index; i < index + 4; i++) {
                        one_row.push(_month[i]);
                    }
                    arrs.push(one_row);
                }
            });
            arrs.forEach(function (_week, index) {
                var cells = _lodash2.default.map(_week, function (_v, _index) {
                    return _react2.default.createElement(
                        Cell,
                        {
                            className: (0, _classnames2.default)({
                                now: _this2.checkIsNow(_v - 1),
                                selected: _this2.checkSelected(_v)
                            }),
                            key: _index,
                            onClick: function onClick() {
                                _this2.cellClick(_v);
                            } },
                        _v,
                        '\u6708'
                    );
                });

                component.push(_react2.default.createElement(
                    'div',
                    { key: index, className: 'mcds-datepicker__calendar-months' },
                    cells
                ));
            });
            return component;
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                _base.DatePickerMonthCalendar,
                null,
                this.renderMonths()
            );
        }
    }]);
    return monthCalendar;
}(_react2.default.Component);

exports.default = monthCalendar;


monthCalendar.propTypes = {
    className: _propTypes2.default.string,
    cellChanged: _propTypes2.default.func,
    init: _propTypes2.default.oneOfType([_propTypes2.default.object, _propTypes2.default.number])
};

monthCalendar.defaultProps = {
    init: Date.parse(new Date()),
    cellChanged: function cellChanged() {}
};
module.exports = exports['default'];
