// carbon/lib/components/date-picker/flavor/base/year-calendar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _base = require('../_base');

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Cell = function Cell(props) {
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, props, { className: (0, _classnames2.default)('mcds-datepicker__calendar-year-oneyear', props.className) }),
        props.children
    );
};

Cell.propTypes = {
    children: _propTypes2.default.any.isRequired,
    className: _propTypes2.default.string
};

var YearCalendar = function (_React$Component) {
    (0, _inherits3.default)(YearCalendar, _React$Component);

    function YearCalendar(props) {
        (0, _classCallCheck3.default)(this, YearCalendar);

        var _this = (0, _possibleConstructorReturn3.default)(this, (YearCalendar.__proto__ || (0, _getPrototypeOf2.default)(YearCalendar)).call(this, props));

        var init = {};

        if (typeof props.init === 'number') {
            var _init = (0, _moment2.default)(props.init);
            init = {
                year: _init.year()
            };
        } else {
            init = props.init;
        }

        _this.cellClick = _this.cellClick.bind(_this);
        _this._checkNow = _this._checkNow.bind(_this);
        _this._checkSelected = _this._checkSelected.bind(_this);
        _this.state = {
            init: init
        };
        return _this;
    }

    (0, _createClass3.default)(YearCalendar, [{
        key: '_getYearInfo',
        value: function _getYearInfo() {
            var startYear = 1899;
            var endYear = 2100;
            return {
                from: startYear,
                to: endYear
            };
        }
    }, {
        key: 'cellClick',
        value: function cellClick(year) {
            this.props.cellChanged({ year: year });
        }
    }, {
        key: '_checkSelected',
        value: function _checkSelected(year) {
            if (year === this.state.init.year) {
                return true;
            }
            return false;
        }
    }, {
        key: '_checkNow',
        value: function _checkNow(year) {
            if (year === (0, _moment2.default)().year()) {
                return true;
            }
            return false;
        }
    }, {
        key: 'renderYears',
        value: function renderYears() {
            var _this2 = this;

            var component = [];
            var arrs = [];
            var years_range = this._getYearInfo();
            var from = years_range.from,
                to = years_range.to;

            for (var i = to; i > from; i--) {
                if (i % 5 === 0) {
                    var _arr = [];
                    for (var index = i; index > i - 5; index--) {
                        _arr.push(index);
                    }
                    arrs.push(_arr);
                }
            }
            arrs.forEach(function (v, index) {

                var cells = _lodash2.default.map(v, function (_v, _index) {
                    return _react2.default.createElement(
                        Cell,
                        {
                            onClick: function onClick() {
                                return _this2.cellClick(_v);
                            },
                            className: (0, _classnames2.default)({
                                now: _this2._checkNow(_v),
                                selected: _this2._checkSelected(_v)
                            }),
                            key: _index },
                        _v
                    );
                });

                component.push(_react2.default.createElement(
                    'div',
                    { key: index, className: 'mcds-datepicker__calendar-years' },
                    cells
                ));
            });
            return component;
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                _base.DatePickerYearCalendar,
                null,
                this.renderYears()
            );
        }
    }]);
    return YearCalendar;
}(_react2.default.Component);

exports.default = YearCalendar;


YearCalendar.propTypes = {
    children: _propTypes2.default.any,
    className: _propTypes2.default.string,
    cellChanged: _propTypes2.default.func,
    init: _propTypes2.default.oneOfType([_propTypes2.default.object, _propTypes2.default.number])
};

YearCalendar.defaultProps = {
    init: Date.parse(new Date()),
    cellChanged: function cellChanged() {}
};
module.exports = exports['default'];
