// carbon/lib/components/buttons/flavors/stateful/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var StatefulButton = function (_React$Component) {
    (0, _inherits3.default)(StatefulButton, _React$Component);

    function StatefulButton() {
        (0, _classCallCheck3.default)(this, StatefulButton);
        return (0, _possibleConstructorReturn3.default)(this, (StatefulButton.__proto__ || (0, _getPrototypeOf2.default)(StatefulButton)).call(this));
    }

    (0, _createClass3.default)(StatefulButton, [{
        key: 'render',
        value: function render() {
            var _props = this.props,
                className = _props.className,
                icon = _props.icon,
                hoverIcon = _props.hoverIcon,
                others = (0, _objectWithoutProperties3.default)(_props, ['className', 'icon', 'hoverIcon']);

            return _react2.default.createElement(
                'button',
                (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-button mcds-button__follow', className) }),
                _react2.default.createElement('i', { className: (0, _classnames2.default)('mcds-stateful__selected mcds-icon__left', icon) }),
                _react2.default.createElement('i', { className: (0, _classnames2.default)('mcds-stateful__not-selected mcds-icon__left', hoverIcon) }),
                this.props.children
            );
        }
    }]);
    return StatefulButton;
}(_react2.default.Component);

StatefulButton.defaultProps = {};
exports.default = StatefulButton;


StatefulButton.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any,
    icon: _propTypes2.default.string,
    hoverIcon: _propTypes2.default.string
};
module.exports = exports['default'];
