// carbon/lib/components/notification/flavors/stack/stack-notifications.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _assign = require('babel-runtime/core-js/object/assign');

var _assign2 = _interopRequireDefault(_assign);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _immutable = require('immutable');

var _notification = require('./notification');

var _notification2 = _interopRequireDefault(_notification);

var _portal = require('../../../../base-components/portal');

var _portal2 = _interopRequireDefault(_portal);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function defaultStyleFactory(index, style) {
    return (0, _assign2.default)({}, style, { top: 1 + index * 4 + 'rem' });
} /* eslint-disable react/jsx-no-bind */

var NotificationStack = function (_Component) {
    (0, _inherits3.default)(NotificationStack, _Component);

    function NotificationStack(props) {
        (0, _classCallCheck3.default)(this, NotificationStack);

        var _this = (0, _possibleConstructorReturn3.default)(this, (NotificationStack.__proto__ || (0, _getPrototypeOf2.default)(NotificationStack)).call(this, props));

        _this.state = {
            notifications: (0, _immutable.OrderedSet)(),
            count: 0
        };
        _this.add = _this.add.bind(_this);
        _this.del = _this.del.bind(_this);
        return _this;
    }

    (0, _createClass3.default)(NotificationStack, [{
        key: 'add',
        value: function add(notify) {
            var _state = this.state,
                notifications = _state.notifications,
                count = _state.count;

            var newCount = count + 1;
            notify.key = newCount;
            return this.setState({
                count: newCount,
                notifications: notifications.add(notify)
            });
        }
    }, {
        key: 'del',
        value: function del(notify) {
            var notifications = this.state.notifications;
            this.setState({ notifications: notifications.delete(notify) });
        }
    }, {
        key: 'render',
        value: function render() {
            var _this2 = this;

            var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : this.props;

            return _react2.default.createElement(
                _portal2.default,
                { isOpened: true },
                _react2.default.createElement(
                    'div',
                    { className: 'notification-list' },
                    this.state.notifications.toArray().map(function (notification, index) {
                        var isLast = index === 0 && _this2.state.notifications.length === 1;
                        // const dismissNow = isLast || !props.dismissInOrder;

                        // Handle styles
                        var barStyle = props.barStyleFactory(index, notification.barStyle);
                        var activeBarStyle = props.activeBarStyleFactory(index, notification.activeBarStyle);

                        var dismissAfter = notification.dismissAfter;


                        if (dismissAfter !== false) {
                            if (dismissAfter) {
                                dismissAfter = notification.dismissAfter;
                            }
                            var beforeDismiss = 0;
                            for (var i = 0; i < index; i++) {
                                beforeDismiss += _this2.state.notifications.toArray()[i].dismissAfter;
                            }
                            dismissAfter += beforeDismiss;
                        }

                        return _react2.default.createElement(_notification2.default, (0, _extends3.default)({}, notification, {
                            key: notification.key,
                            isLast: isLast,
                            message: '' + notification.message,
                            action: notification.action || props.action,
                            dismissAfter: dismissAfter,
                            className: 'mcds-notification__toast ' + props.className,
                            onDismiss: function onDismiss() {
                                _this2.del(notification);
                            },
                            activeBarStyle: activeBarStyle,
                            barStyle: barStyle }));
                    })
                )
            );
        }
    }]);
    return NotificationStack;
}(_react.Component);

NotificationStack.propTypes = {
    activeBarStyleFactory: _propTypes2.default.func,
    barStyleFactory: _propTypes2.default.func,
    className: _propTypes2.default.string,
    dismissInOrder: _propTypes2.default.bool,
    onClick: _propTypes2.default.func,
    action: _propTypes2.default.string
};

NotificationStack.defaultProps = {
    activeBarStyleFactory: defaultStyleFactory,
    barStyleFactory: defaultStyleFactory,
    dismissInOrder: true,
    onClick: function onClick() {}
};

exports.default = NotificationStack;
module.exports = exports['default'];
