// carbon/lib/components/date-picker/index.example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./index');

var _form = require('../form');

var _reactMarkdown = require('react-markdown');

var _reactMarkdown2 = _interopRequireDefault(_reactMarkdown);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DatePickerExampleCode = '\nclass Example extends React.Component {\n    ...\n    onChanged(v) {\n        console.log(\'onChanged\',v);\n    }\n    render() {\n        return (\n            <DatePicker onChanged={this.onChanged.bind(this)} >\n                <Button className="mcds-button__brand">\n                    \u65F6\u95F4\u9009\u62E9\u5668\n                </Button>\n            </DatePicker>\n        );\n    }\n}';

var DatePickerExample = function (_React$Component) {
    (0, _inherits3.default)(DatePickerExample, _React$Component);

    function DatePickerExample() {
        (0, _classCallCheck3.default)(this, DatePickerExample);
        return (0, _possibleConstructorReturn3.default)(this, (DatePickerExample.__proto__ || (0, _getPrototypeOf2.default)(DatePickerExample)).apply(this, arguments));
    }

    (0, _createClass3.default)(DatePickerExample, [{
        key: 'onChange',
        value: function onChange(v) {
            console.log('onChanged', v);
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                _index.DatePicker,
                { error: true, onChange: this.onChange.bind(this), closeOnOutsideClick: true },
                _react2.default.createElement(_form.Input, {
                    iconRight: _react2.default.createElement('span', { onClick: function onClick() {
                            console.log('span click');
                        }, className: 'mcds-icon__data-line-20' }),
                    ref: 'input' })
            );
        }
    }]);
    return DatePickerExample;
}(_react2.default.Component);

var Intro = '##### DatePicker提供以下参数\r\n\r\n- DatePicker\r\n\t- init 初始化时间戳(默认为当前时间点)\r\n\t- onChanged 结束回调\r\n\t- onClose 关闭时候的回调\t\r\n\t- onOpen 展示这前回调\r\n\t- placement 打开的位置\r\n';
exports.default = [{
    id: 'datePicker',
    element: _react2.default.createElement(DatePickerExample, null),
    intro: _react2.default.createElement(_reactMarkdown2.default, { className: 'markdown', source: Intro }),
    code: DatePickerExampleCode
}];
module.exports = exports['default'];
