// carbon/lib/components/form/flavors/select/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _dropdown = require('../../../dropdown');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var isAvailable = function isAvailable(param) {
    var isUnavailable = _lodash2.default.isNull(param) || _lodash2.default.isUndefined(param);
    return !isUnavailable;
};

var Select = function (_React$Component) {
    (0, _inherits3.default)(Select, _React$Component);

    function Select(props) {
        (0, _classCallCheck3.default)(this, Select);

        // 这里干掉buildDate是考虑到效率问题,每次build挺慢的
        var _this = (0, _possibleConstructorReturn3.default)(this, (Select.__proto__ || (0, _getPrototypeOf2.default)(Select)).call(this, props));

        _this.state = {
            // 不要在这里加入data.更新问题会很多
            display: 'none',
            defaultValue: isAvailable(_this.props.value) ? _this.getDefaultValue(_this.props.value) : '请选择',
            value: typeof _this.props.value !== 'undefined' ? _this.props.value : '-1'
        };
        return _this;
    }

    (0, _createClass3.default)(Select, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            this.mounted = true;
            var active = this.props.active;

            if (active === true && this.refs.select__wrap) {
                this.refs.select__wrap.focus();
            }
        }
    }, {
        key: 'componentWillReceiveProps',
        value: function componentWillReceiveProps(nextProps) {
            // 干掉的原因,是为了select 显示的内容并非是option显示的内容,
            var defaultValue = typeof nextProps.value !== 'undefined' ? nextProps.value : '请选择';
            // if (defaultValue == this.state.defaultValue) {
            this.setState({ defaultValue: defaultValue });
            // }
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            this.mounted = false;
        }

        /**
        * 用来计算显示出来的文字 .
        * if defaultValue in children
        * value = child
        * else
        * value = props.defaultValue
        * @param  {[type]} value [description]
        * @return {[type]}       [description]
        */

    }, {
        key: 'getDefaultValue',
        value: function getDefaultValue(defaultValue) {
            var child = void 0;
            _react2.default.Children.forEach(this.props.children, function (v) {
                if (v.props.value === defaultValue) {
                    child = v.props.children;
                }
            });
            if (!child) {
                child = defaultValue;
            }
            return child;
        }
    }, {
        key: '_handleOnChick',
        value: function _handleOnChick(obj) {
            if (this.mounted) {
                this.setState({
                    defaultValue: obj.props.children,
                    value: obj.props.value || obj.props.children
                });
                var select = this.refs.select;
                if (select && select.close) {
                    this.refs.select.close();
                }
            }
            this.props.onChange(obj.props.value || obj.props.children);
        }
    }, {
        key: 'buildOptions',
        value: function buildOptions() {
            var _this2 = this;

            var options = [];
            _react2.default.Children.forEach(this.props.children, function (v, index) {
                var isSelected = v.props.value === _this2.state.value;
                if (typeof v.props.value === 'undefined') {
                    // 如果没有传入value的话.用children去做判断
                    isSelected = String(v.props.children) === String(_this2.state.value);
                }
                var tmp = _react2.default.createElement(
                    _dropdown.DropDownItem,
                    { isSelected: isSelected, onClick: _this2._handleOnChick.bind(_this2, v), className: 'close', key: index, iconLeft: isSelected ? CheckBox : CheckBoxPlace },
                    v.props.children
                );
                options.push(tmp);
            });

            return _react2.default.createElement(
                _dropdown.DropDown,
                { className: (0, _classnames2.default)('mcds-select__options') },
                _react2.default.createElement(
                    _dropdown.DropDownList,
                    null,
                    options
                )
            );
        }
    }, {
        key: 'renderDisable',
        value: function renderDisable() {
            return _react2.default.createElement(
                'div',
                { className: 'mcds-select__wrap mcds-select__disabled' },
                this.state.defaultValue
            );
        }
    }, {
        key: 'render',
        value: function render() {
            var _this3 = this;

            var options = this.buildOptions();
            var _props = this.props,
                placement = _props.placement,
                target = _props.target,
                error = _props.error;

            var requiredSpan = this.props.required ? _react2.default.createElement(
                'span',
                { className: 'mcds-span__required' },
                '*'
            ) : null;
            var selectLabel = this.props.label ? _react2.default.createElement(
                'label',
                { className: 'mcds-label', htmlFor: this.props.id },
                this.props.label
            ) : null;
            return _react2.default.createElement(
                'div',
                { className: (0, _classnames2.default)('mcds-element__container', this.props.className), ref: function ref(node) {
                        _this3.node = node;
                    } },
                requiredSpan,
                ' ',
                selectLabel,
                this.props.disabled ? this.renderDisable() : _react2.default.createElement(
                    _dropdown.DropDownTrigger,
                    {
                        ref: 'select',
                        target: target,
                        offset: 5,
                        synchWidth: true,
                        placement: placement,
                        className: (0, _classnames2.default)('mcds-select__control', { 'mcds-element__border': error }) },
                    _react2.default.createElement(
                        'div',
                        { ref: 'select__wrap', tabIndex: '0', className: 'mcds-select__wrap' },
                        this.state.defaultValue
                    ),
                    options
                )
            );
        }
    }]);
    return Select;
}(_react2.default.Component);

Select.propTypes = {
    id: _propTypes2.default.string,
    value: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.number]),
    target: _propTypes2.default.string,
    label: _propTypes2.default.string,
    className: _propTypes2.default.string,
    placement: _propTypes2.default.string,
    children: _propTypes2.default.any,
    disabled: _propTypes2.default.bool,
    onChange: _propTypes2.default.func,
    required: _propTypes2.default.bool,
    error: _propTypes2.default.bool,
    active: _propTypes2.default.bool
};
// const iconLeft = <span className="mcds-icon__left mcds-icon__check-mark-small" />;
var CheckBox = _react2.default.createElement('span', { className: 'mcds-icon__left mcds-icon__checked mcds-dropdown__checkbox mcds-icon__check-12' });
var CheckBoxPlace = _react2.default.createElement('span', { className: 'mcds-icon__left mcds-icon__checked mcds-dropdown__checkbox mcds-dropdown__checkbox-place' });

Select.defaultProps = {
    target: 'body',
    onChange: function onChange() {},
    placement: 'bottom-right'
};

exports.default = Select;
module.exports = exports['default'];
