// carbon/lib/components/form/flavors/checkbox-add-button/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CheckboxAddButton = function (_React$Component) {
    (0, _inherits3.default)(CheckboxAddButton, _React$Component);

    function CheckboxAddButton(props) {
        (0, _classCallCheck3.default)(this, CheckboxAddButton);

        var _this = (0, _possibleConstructorReturn3.default)(this, (CheckboxAddButton.__proto__ || (0, _getPrototypeOf2.default)(CheckboxAddButton)).call(this, props));

        _this.state = {
            checked: !!_this.props.checked
        };
        return _this;
    }

    (0, _createClass3.default)(CheckboxAddButton, [{
        key: 'componentWillReceiveProps',
        value: function componentWillReceiveProps(nextProps) {
            this.setState({ checked: !!nextProps.checked });
        }
    }, {
        key: 'handleChange',
        value: function handleChange(e) {
            this.setState({ checked: e.target.checked });
            this.props.callback(e.target.checked);
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'div',
                { className: (0, _classnames2.default)('mcds-checkbox__add', this.props.className) },
                _react2.default.createElement('input', {
                    className: 'mcds-assistive__text',
                    type: 'checkbox',
                    id: this.props.id,
                    name: this.props.name,
                    checked: this.state.checked,
                    disabled: this.props.disabled,
                    onChange: this.handleChange.bind(this) }),
                _react2.default.createElement(
                    'label',
                    { htmlFor: this.props.id, className: 'mcds-checkbox__faux' },
                    _react2.default.createElement(
                        'span',
                        { className: 'mcds-assistive__text' },
                        this.props.label
                    )
                )
            );
        }
    }]);
    return CheckboxAddButton;
}(_react2.default.Component);

CheckboxAddButton.defaultProps = {
    callback: function callback(val) {
        console.log(val);
        return val;
    }
};


CheckboxAddButton.propTypes = {
    id: _propTypes2.default.string,
    name: _propTypes2.default.string.isRequired,
    label: _propTypes2.default.string.isRequired,
    className: _propTypes2.default.string,
    checked: _propTypes2.default.bool,
    disabled: _propTypes2.default.bool,
    callback: _propTypes2.default.func
};

exports.default = CheckboxAddButton;
module.exports = exports['default'];
