// carbon/lib/components/form/flavors/radio-group-alternate/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AlternateRadio = function (_React$Component) {
    (0, _inherits3.default)(AlternateRadio, _React$Component);

    function AlternateRadio(props) {
        (0, _classCallCheck3.default)(this, AlternateRadio);

        var _this = (0, _possibleConstructorReturn3.default)(this, (AlternateRadio.__proto__ || (0, _getPrototypeOf2.default)(AlternateRadio)).call(this, props));

        _this.state = {
            checked: !!_this.props.checked
        };
        return _this;
    }

    (0, _createClass3.default)(AlternateRadio, [{
        key: 'handleChange',
        value: function handleChange(e) {
            this.props.callback(e.target.checked);
        }
    }, {
        key: 'render',
        value: function render() {
            return _react2.default.createElement(
                'span',
                { className: (0, _classnames2.default)('mcds-radio__button', this.props.className) },
                _react2.default.createElement(
                    'label',
                    { className: 'mcds-radio-button__label' },
                    _react2.default.createElement('input', {
                        name: this.props.name,
                        type: 'radio',
                        id: this.props.id,
                        checked: this.props.checked,
                        disabled: this.props.disabled,
                        onClick: this.handleChange.bind(this) }),
                    _react2.default.createElement(
                        'span',
                        { className: 'mcds-radio__faux' },
                        this.props.children
                    )
                )
            );
        }
    }]);
    return AlternateRadio;
}(_react2.default.Component);

AlternateRadio.defaultProps = {
    callback: function callback(val) {
        return val;
    }
};


AlternateRadio.propTypes = {
    id: _propTypes2.default.string,
    name: _propTypes2.default.string,
    className: _propTypes2.default.string,
    disabled: _propTypes2.default.string,
    checked: _propTypes2.default.string,
    children: _propTypes2.default.any,
    callback: _propTypes2.default.func
};

var AlternateGroup = function AlternateGroup(_ref) {
    var className = _ref.className,
        children = _ref.children,
        label = _ref.label;
    return _react2.default.createElement(
        'div',
        { className: (0, _classnames2.default)(className) },
        _react2.default.createElement(
            'div',
            { className: 'mcds-form-element__label' },
            label
        ),
        _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(
                'div',
                { className: 'mcds-radio-button__group' },
                children
            )
        )
    );
};

AlternateGroup.propTypes = {
    label: _propTypes2.default.string,
    className: _propTypes2.default.string,
    children: _propTypes2.default.any
};

exports.default = {
    AlternateRadio: AlternateRadio,
    AlternateGroup: AlternateGroup
};
module.exports = exports['default'];
