// carbon/lib/components/tab/flavors/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Tab = function Tab(_ref) {
    var className = _ref.className,
        children = _ref.children;
    return _react2.default.createElement(
        'div',
        { className: className },
        _react2.default.createElement(
            'ul',
            { className: 'mcds-tab__items mcds-list__horizontal' },
            _react2.default.Children.map(children, function (element) {
                if (element.type === TabItem) {
                    return element;
                }
                return null;
            })
        ),
        _react2.default.Children.map(children, function (element) {
            if (element.type === TabContent) {
                return element;
            }
        })
    );
};

Tab.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any,
    elements: function elements(props) {
        var typeList = ['Tab', 'TabItem', 'TabContent'];
        var flag = false;
        props.children.forEach(function (element) {
            if (typeof element.type === 'string') {
                flag = true;
            }
        });
        if (flag) {
            return new Error('Tab children must one of ' + typeList);
        }
    }
};

var TabItem = function TabItem(_ref2) {
    var className = _ref2.className,
        children = _ref2.children,
        others = (0, _objectWithoutProperties3.default)(_ref2, ['className', 'children']);
    return _react2.default.createElement(
        'li',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-tab__item mcds-list__item', className) }),
        children
    );
};

TabItem.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any
};

var TabContent = function TabContent(_ref3) {
    var className = _ref3.className,
        children = _ref3.children,
        others = (0, _objectWithoutProperties3.default)(_ref3, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-content__item', className) }),
        children
    );
};

TabContent.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any
};

exports.default = {
    Tab: Tab,
    TabItem: TabItem,
    TabContent: TabContent
};
module.exports = exports['default'];
