// carbon/lib/components/navigation/flavors/vertical/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.NavItem = exports.NavList = exports.NavTitle = exports.Navigation = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Navigation = function Navigation(_ref) {
    var className = _ref.className,
        children = _ref.children,
        others = (0, _objectWithoutProperties3.default)(_ref, ['className', 'children']);
    return _react2.default.createElement(
        'div',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-navigation-list__vertical', className) }),
        children
    );
};

exports.Navigation = Navigation;
Navigation.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any
};

var NavTitle = function NavTitle(_ref2) {
    var id = _ref2.id,
        className = _ref2.className,
        children = _ref2.children,
        others = (0, _objectWithoutProperties3.default)(_ref2, ['id', 'className', 'children']);
    return _react2.default.createElement(
        'h2',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-text-title__caps mcds-p-around__medium', className), id: id }),
        children
    );
};

exports.NavTitle = NavTitle;
NavTitle.propTypes = {
    id: _propTypes2.default.string,
    className: _propTypes2.default.string,
    children: _propTypes2.default.any
};

var NavList = function NavList(_ref3) {
    var className = _ref3.className,
        children = _ref3.children,
        others = (0, _objectWithoutProperties3.default)(_ref3, ['className', 'children']);
    return _react2.default.createElement(
        'ul',
        (0, _extends3.default)({}, others, { className: className }),
        children
    );
};

exports.NavList = NavList;
NavList.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any
};

var NavItem = function NavItem(_ref4) {
    var id = _ref4.id,
        className = _ref4.className,
        children = _ref4.children,
        others = (0, _objectWithoutProperties3.default)(_ref4, ['id', 'className', 'children']);
    return _react2.default.createElement(
        'li',
        (0, _extends3.default)({}, others, { className: className }),
        _react2.default.createElement(
            'span',
            { className: 'mcds-navigation-list__vertical-action mcds-text-link__reset', id: id },
            children
        )
    );
};

exports.NavItem = NavItem;
NavItem.propTypes = {
    id: _propTypes2.default.string,
    className: _propTypes2.default.string,
    children: _propTypes2.default.any
};
