// carbon/lib/components/editor/flavors/base/src/plugins/color/panel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _keys = require('babel-runtime/core-js/object/keys');

var _keys2 = _interopRequireDefault(_keys);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _index = require('./index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Panel = function (_Component) {
    (0, _inherits3.default)(Panel, _Component);

    function Panel(props) {
        (0, _classCallCheck3.default)(this, Panel);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Panel.__proto__ || (0, _getPrototypeOf2.default)(Panel)).call(this, props));

        _this.state = { currentColor: '' };
        return _this;
    }

    (0, _createClass3.default)(Panel, [{
        key: 'setCurrentColor',
        value: function setCurrentColor(color) {
            if (color !== this.state.currentColor) {
                this.setState({ currentColor: color });
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var _this2 = this;

            var onClick = this.props.onClick;
            var currentColor = this.state.currentColor;

            return _react2.default.createElement(
                'div',
                { className: 'mcds-editor__color-root' },
                (0, _keys2.default)(_index.COLOR_MAP).map(function (c) {
                    return _react2.default.createElement(PanelItem, {
                        active: currentColor === c,
                        key: 'color' + c,
                        color: _index.COLOR_MAP[c].color,
                        onClick: onClick.bind(_this2, c) });
                })
            );
        }
    }]);
    return Panel;
}(_react.Component);

exports.default = Panel;

Panel.propTypes = {
    onClick: _propTypes2.default.func
};

var PanelItem = function PanelItem(_ref) {
    var color = _ref.color,
        onClick = _ref.onClick,
        active = _ref.active;
    return _react2.default.createElement('span', {
        className: (0, _classnames2.default)('mcds-editor__color-btn', { 'mcds-editor__color-active': active }),
        style: { backgroundColor: color },
        onClick: onClick });
};
PanelItem.propTypes = {
    color: _propTypes2.default.string,
    active: _propTypes2.default.bool,
    onClick: _propTypes2.default.func
};
module.exports = exports['default'];
