// carbon/lib/components/form/flavors/input-required/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Input = function (_Component) {
    (0, _inherits3.default)(Input, _Component);

    function Input(props) {
        (0, _classCallCheck3.default)(this, Input);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Input.__proto__ || (0, _getPrototypeOf2.default)(Input)).call(this, props));

        _this.state = {
            requiredFlag: false
        };
        return _this;
    }

    (0, _createClass3.default)(Input, [{
        key: 'requiredBlur',
        value: function requiredBlur() {
            var val = this.props.validFun(this.node.value);
            this.setState({ requiredFlag: !!val });
        }
    }, {
        key: 'handleOnChage',
        value: function handleOnChage() {
            this.props.onChange(this.node.value);
        }
    }, {
        key: '_renderErrorSpan',
        value: function _renderErrorSpan() {
            var result = null;
            if (this.props.errorMsg && this.state.requiredFlag) {
                result = _react2.default.createElement(
                    'span',
                    { className: 'mcds-span__required' },
                    _react2.default.createElement('i', { className: this.props.icon }),
                    this.props.errorMsg
                );
            }
            return result;
        }
    }, {
        key: 'render',
        value: function render() {
            var _this2 = this;

            var requireInput = this.state.requiredFlag ? 'mcds-input__required' : '';
            var error = this.props.error;

            return _react2.default.createElement(
                'div',
                { className: (0, _classnames2.default)('mcds-input__container', { 'mcds-input__border': error }) },
                this.props.label ? _react2.default.createElement(
                    'label',
                    { className: 'mcds-label' },
                    _react2.default.createElement(
                        'span',
                        { className: 'mcds-span__required' },
                        '*'
                    ),
                    ' ',
                    this.props.label
                ) : null,
                _react2.default.createElement('input', {
                    id: this.props.id,
                    className: (0, _classnames2.default)('mcds-input', requireInput),
                    name: this.props.name, type: 'text',
                    ref: function ref(node) {
                        _this2.node = node;
                    },
                    autoComplete: 'off',
                    placeholder: this.props.placeholder,
                    required: this.props.required,
                    onBlur: this.requiredBlur.bind(this),
                    onChange: this.handleOnChage.bind(this),
                    defaultValue: this.props.value }),
                this._renderErrorSpan()
            );
        }
    }]);
    return Input;
}(_react.Component);

Input.propTypes = {
    id: _propTypes2.default.string,
    icon: _propTypes2.default.string,
    errorMsg: _propTypes2.default.string,
    label: _propTypes2.default.string,
    placeholder: _propTypes2.default.string,
    name: _propTypes2.default.string.isRequired,
    validFun: _propTypes2.default.func,
    required: _propTypes2.default.string,
    value: _propTypes2.default.string,
    onChange: _propTypes2.default.func,
    error: _propTypes2.default.bool
};
Input.defaultProps = {
    name: 'name',
    required: null,
    validFun: function validFun(val) {
        return val.trim() === '';
    },
    onChange: function onChange() {}
};
exports.default = Input;
module.exports = exports['default'];
