// carbon/lib/components/images/figure/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Image = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _index = require('./../../button-group/index');

var _index2 = require('./../../button-icon/index');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Image = function Image(_ref) {
    var className = _ref.className,
        cropClass = _ref.cropClass,
        titleClass = _ref.titleClass,
        image = _ref.image,
        actions = _ref.actions,
        title = _ref.title,
        imgSrc = _ref.imgSrc,
        imgIcon = _ref.imgIcon,
        iconClass = _ref.iconClass,
        others = (0, _objectWithoutProperties3.default)(_ref, ['className', 'cropClass', 'titleClass', 'image', 'actions', 'title', 'imgSrc', 'imgIcon', 'iconClass']);


    var imageContent = _react2.default.createElement('img', { src: imgSrc, alt: 'Description of the image' });
    var actionsContent = _react2.default.createElement(
        _index.ButtonGroup,
        { className: 'mcds-image__actions' },
        _react2.default.createElement(_index2.ButtonIcon, { className: 'mcds-button__item', icon: 'mcds-icon__download-line-20' }),
        _react2.default.createElement(_index2.ButtonIcon, { className: 'mcds-button__item', icon: 'mcds-icon__triangle-solid-14' })
    );

    var imgIconContent = _react2.default.createElement(
        'span',
        { className: 'mcds-icon_container' },
        _react2.default.createElement('span', { className: (0, _classnames2.default)('mcds-icon mcds-icon__size-20 mcds-icon__left', imgIcon) })
    );

    if (image) {
        imageContent = _react2.default.createElement('img', { src: imgSrc, alt: 'Description of the image' });
    } else {
        imageContent = _react2.default.createElement(
            'span',
            { className: 'mcds-image__icon' },
            _react2.default.createElement('span', { className: (0, _classnames2.default)('mcds-icon', imgIcon, iconClass) }),
            _react2.default.createElement(
                'span',
                { className: 'mcds-assistive-text' },
                title || 'Image Title'
            )
        );
    }

    return _react2.default.createElement(
        'figure',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-image', className) }),
        _react2.default.createElement(
            'a',
            { href: 'javascript:void(0);', className: (0, _classnames2.default)('mcds-image__crop', cropClass) },
            imageContent
        ),
        actions ? actionsContent : null,
        _react2.default.createElement(
            'figcaption',
            { className: (0, _classnames2.default)('mcds-image__title', titleClass) },
            imgIcon ? imgIconContent : null,
            _react2.default.createElement(
                'span',
                { className: 'mcds-image__text mcds-truncate', title: title || 'Image Title' },
                title || 'Image Title'
            )
        )
    );
};

exports.Image = Image;
Image.propTypes = {
    className: _propTypes2.default.string,
    cropClass: _propTypes2.default.string,
    titleClass: _propTypes2.default.string,
    image: _propTypes2.default.bool,
    actions: _propTypes2.default.bool,
    title: _propTypes2.default.string,
    imgSrc: _propTypes2.default.string,
    imgIcon: _propTypes2.default.string,
    iconClass: _propTypes2.default.string
};
