// carbon/lib/components/form/flavors/radio/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Radio = function (_React$Component) {
    (0, _inherits3.default)(Radio, _React$Component);

    function Radio() {
        (0, _classCallCheck3.default)(this, Radio);
        return (0, _possibleConstructorReturn3.default)(this, (Radio.__proto__ || (0, _getPrototypeOf2.default)(Radio)).call(this));
    }

    (0, _createClass3.default)(Radio, [{
        key: 'handleOnChange',
        value: function handleOnChange() {
            this.props.onChange(this.node.value);
        }
    }, {
        key: 'render',
        value: function render() {
            var _this2 = this;

            var _props = this.props,
                id = _props.id,
                name = _props.name,
                className = _props.className,
                disabled = _props.disabled,
                checked = _props.checked,
                label = _props.label,
                others = (0, _objectWithoutProperties3.default)(_props, ['id', 'name', 'className', 'disabled', 'checked', 'label']);

            return _react2.default.createElement(
                'label',
                { className: (0, _classnames2.default)(className, 'mcds-radio__block') },
                _react2.default.createElement(
                    'span',
                    { className: 'mcds-radio__checked' },
                    _react2.default.createElement('input', (0, _extends3.default)({}, others, {
                        id: id,
                        ref: function ref(node) {
                            _this2.node = node;
                        },
                        onChange: this.handleOnChange.bind(this),
                        className: 'mcds-radio',
                        type: 'radio',
                        name: name,
                        disabled: disabled,
                        defaultChecked: checked })),
                    _react2.default.createElement('span', { className: 'mcds-radio__inner' })
                ),
                label
            );
        }
    }]);
    return Radio;
}(_react2.default.Component);

Radio.defaultProps = {
    onChange: function onChange() {}
};


Radio.propTypes = {
    id: _propTypes2.default.string,
    name: _propTypes2.default.string,
    label: _propTypes2.default.any,
    disabled: _propTypes2.default.bool,
    checked: _propTypes2.default.bool,
    className: _propTypes2.default.string,
    onChange: _propTypes2.default.func
};

var RadioGroup = function RadioGroup(_ref) {
    var groupLable = _ref.groupLable,
        children = _ref.children,
        className = _ref.className;
    return _react2.default.createElement(
        'div',
        { className: className },
        _react2.default.createElement(
            'span',
            { className: 'mcds-radio__group-label' },
            groupLable
        ),
        _react2.default.createElement(
            'div',
            null,
            children
        )
    );
};

RadioGroup.propTypes = {
    groupLable: _propTypes2.default.string,
    children: _propTypes2.default.any,
    className: _propTypes2.default.string
};

var RequiredRadioGroup = function RequiredRadioGroup(_ref2) {
    var groupLable = _ref2.groupLable,
        children = _ref2.children,
        errorMsg = _ref2.errorMsg,
        className = _ref2.className;
    return _react2.default.createElement(
        'div',
        { className: className },
        _react2.default.createElement(
            'span',
            { className: 'mcds-radio__group-label' },
            _react2.default.createElement(
                'span',
                { className: 'mcds-span__required' },
                '*'
            ),
            groupLable
        ),
        _react2.default.createElement(
            'div',
            null,
            children,
            _react2.default.createElement(
                'span',
                { className: 'mcds-span__required' },
                errorMsg
            )
        )
    );
};

RequiredRadioGroup.propTypes = {
    groupLable: _propTypes2.default.string,
    errorMsg: _propTypes2.default.string,
    children: _propTypes2.default.any,
    className: _propTypes2.default.string
};

exports.default = {
    Radio: Radio,
    RadioGroup: RadioGroup,
    RequiredRadioGroup: RequiredRadioGroup
};
module.exports = exports['default'];
