// carbon/lib/components/editor/flavors/base/src/utils/create-chained-function.test.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = test;

var _assert = require('assert');

var _assert2 = _interopRequireDefault(_assert);

var _createChainedFunction = require('./create-chained-function');

var _createChainedFunction2 = _interopRequireDefault(_createChainedFunction);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function test() {
    testNull();
    testOneFun();
    testMoreFun();
}

function testNull() {
    (0, _assert2.default)((0, _createChainedFunction2.default)() === null);
    (0, _assert2.default)((0, _createChainedFunction2.default)(null) === null);
}

function testOneFun() {
    var fn1 = function fn1(a, b) {
        return a + b;
    };
    var fn = (0, _createChainedFunction2.default)(fn1);
    _assert2.default.equal(fn, fn1);
}

function testMoreFun() {
    var sum = 0;
    var fn1 = function fn1(a, b) {
        return sum += a + b;
    };
    var fn2 = function fn2(a, b) {
        return sum += a * b;
    };
    var fn = (0, _createChainedFunction2.default)(fn1, 1, '2', /./, null, undefined, [], fn2);
    _assert2.default.equal(sum, 0);
    fn(2, 5);
    _assert2.default.equal(sum, 17);
}
module.exports = exports['default'];
