// carbon/lib/components/form/flavors/textarea-required/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TextAreaRequired = function (_React$Component) {
    (0, _inherits3.default)(TextAreaRequired, _React$Component);

    function TextAreaRequired(props) {
        (0, _classCallCheck3.default)(this, TextAreaRequired);

        var _this = (0, _possibleConstructorReturn3.default)(this, (TextAreaRequired.__proto__ || (0, _getPrototypeOf2.default)(TextAreaRequired)).call(this, props));

        _this.state = {
            requiredFlag: false
        };
        return _this;
    }

    (0, _createClass3.default)(TextAreaRequired, [{
        key: 'requiredBlur',
        value: function requiredBlur() {
            var val = this.props.validFun(this.refs.textarea.value);
            this.setState({ requiredFlag: val });
        }
    }, {
        key: 'handleOnChange',
        value: function handleOnChange() {
            this.props.onChange(this.refs.textarea.value);
        }
    }, {
        key: 'render',
        value: function render() {
            var error = this.props.error;

            var requireTextArea = this.state.requiredFlag ? 'mcds-textarea__required' : '';
            var requiredErrorSpan = this.state.requiredFlag ? _react2.default.createElement(
                'span',
                { className: 'mcds-span__required' },
                _react2.default.createElement('i', { className: this.props.icon }),
                this.props.errorMsg
            ) : '';
            var textareaLabel = this.props.label ? _react2.default.createElement(
                'label',
                { className: 'mcds-label' },
                _react2.default.createElement(
                    'span',
                    { className: 'mcds-span__required' },
                    '*'
                ),
                this.props.label
            ) : null;
            return _react2.default.createElement(
                'div',
                { className: (0, _classnames2.default)('mcds-textarea__container', this.props.className, { 'mcds-textarea__border': error }) },
                textareaLabel,
                _react2.default.createElement('textarea', {
                    id: this.props.id,
                    ref: 'textarea',
                    type: 'text',
                    onChange: this.handleOnChange.bind(this),
                    name: this.props.name,
                    className: (0, _classnames2.default)('mcds-textarea', requireTextArea),
                    placeholder: this.props.placeholder,
                    onBlur: this.requiredBlur.bind(this) }),
                requiredErrorSpan
            );
        }
    }]);
    return TextAreaRequired;
}(_react2.default.Component);

TextAreaRequired.propTypes = {
    id: _propTypes2.default.string.isRequired,
    icon: _propTypes2.default.string,
    className: _propTypes2.default.string,
    errorMsg: _propTypes2.default.string,
    label: _propTypes2.default.string,
    placeholder: _propTypes2.default.string,
    name: _propTypes2.default.string.isRequired,
    validFun: _propTypes2.default.func,
    error: _propTypes2.default.bool,
    onChange: function onChange() {}
};
TextAreaRequired.defaultProps = {
    name: 'name',
    required: null,
    validFun: function validFun(val) {
        return val.trim() === '';
    }
};
exports.default = TextAreaRequired;
module.exports = exports['default'];
