// carbon/lib/components/data-table/flavors/basic/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.Table = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Table = function Table(_ref) {
    var className = _ref.className,
        children = _ref.children,
        others = (0, _objectWithoutProperties3.default)(_ref, ['className', 'children']);
    return _react2.default.createElement(
        'table',
        (0, _extends3.default)({}, others, { className: (0, _classnames2.default)('mcds-table mcds-table__bordered', className) }),
        children
    );
};

/* class Table extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            mousedown: false,
            resizeable: false,
            screenXStart: 0,
            tdWidth: 0,
            headerWidth: 0,
            targetTd: null
        };
    }

    onmousedown(event) {
        if (this.state.resizeable === true) {
            this.setState({
                mousedown: true,
                screenXStart: event.screenX,
                tdWidth: this.state.targetTd.offsetWidth,
                headerWidth: this.refs.table_column.offsetWidth
            });
        }
        event.preventDefault();
    }

    onmousemove(event) {
        let curEle = event.target.parentElement;
        while (curEle.nodeName !== 'TR' && curEle.nodeName !== 'TH') {
            curEle = curEle.parentElement;
        }
        let offsetX = event.clientX - curEle.getBoundingClientRect().left;
        let table = this.refs.table_column;

            let curTr = table.rows[0];
            if (curEle.offsetWidth - offsetX <= 4) {
                this.setState({
                    targetTd: curEle,
                    resizeable: true
                });
            } else if (offsetX <= 4 && curEle.cellIndex > 0) {
                this.setState({
                    targetTd: curTr.cells[curEle.cellIndex - 1],
                    resizeable: true
                });
            } else {
                this.setState({
                    resizeable: false
                });
            }
    }


    onmouseup(event) {
        let table = this.refs.table_column;
        if (this.state.mousedown === true) {
            let width = this.state.tdWidth + (event.screenX - this.state.screenXStart) + 'px';
            let targetEle = this.state.targetTd;
            targetEle.style.width = width;
            table.style.width = this.state.headerWidth + (event.screenX - this.state.screenXStart) + 'px'; // 调整整个table的宽度
        }
        this.setState({
            targetTd: null,
            resizeable: false,
            mousedown: false
        });
        event.preventDefault();
    }

    renderTh(thead) {
        let ths = [];
        let { resize } = this.props;
        React.Children.map(thead, (curTh, index) => {
            if (resize) {
                ths.push(React.cloneElement(curTh, {
                    key: index,
                    onMouseDown: this.onmousedown.bind(this),
                    onMouseMove: this.onmousemove.bind(this),
                    onMouseUp: this.onmouseup.bind(this)
                }));
            } else {
                ths.push(React.cloneElement(curTh, {key: index}));
            }
        });
        return ths;
    }

    render() {
        let {className, children } = this.props;
        return (
            <table ref="table_column" className={classnames('mcds-table mcds-table__bordered', className)}>
                {React.Children.map(children, v => {
                    if (v.type === 'thead') {
                        return <thead><tr>{this.renderTh(v.props.children.props.children)}</tr></thead>;
                    }
                    return v;
                })}
            </table>
        );
    }
} */

exports.Table = Table;
Table.propTypes = {
    className: _propTypes2.default.string,
    children: _propTypes2.default.any,
    resize: _propTypes2.default.bool
};
